diff -ruN newlib-4.2.0/newlib/libc/unix/getlogin.c newlib-4.2.0.new/newlib/libc/unix/getlogin.c
--- newlib-4.2.0/newlib/libc/unix/getlogin.c	2021-12-31 18:46:13.000000000 +0100
+++ newlib-4.2.0.new/newlib/libc/unix/getlogin.c	2023-07-18 09:22:43.719300008 +0200
@@ -16,9 +16,7 @@
   extern char *ttyname ();
   char *tty;
 
-  if (((tty = ttyname (0)) == 0)
-      || ((tty = ttyname (1)) == 0)
-      || ((tty = ttyname (2)) == 0))
+  if ((tty = ttyname (0)) == 0)
     return 0;
 
   if ((utmp_fd = open (UTMP_FILE, O_RDONLY)) == -1)
