diff -ru cpio-2.14/lib/rtapelib.c cpio-2.14.new/lib/rtapelib.c
--- cpio-2.14/lib/rtapelib.c	2023-04-27 10:39:40 +0000
+++ cpio-2.14.new/lib/rtapelib.c	2023-08-04 07:35:32 +0000
@@ -360,8 +360,8 @@
 
   if (!pw)
     return "getpwuid";
-  if (initgroups (pw->pw_name, gid) != 0)
-    return "initgroups";
+/*  if (initgroups (pw->pw_name, gid) != 0)
+    return "initgroups";*/
   if (gid != getegid () && setgid (gid) != 0 && errno != EPERM)
     return "setgid";
   if (uid != geteuid () && setuid (uid) != 0 && errno != EPERM)
@@ -446,11 +446,14 @@
   assume (remote_file);
 
   /* FIXME: Should somewhat validate the decoding, here.  */
+  /*
   if (gethostbyname (remote_host) == NULL)
     error (EXIT_ON_EXEC_ERROR, 0, _("Cannot connect to %s: resolve failed"),
 	   remote_host);
 
   if (remote_user && *remote_user == '\0')
+  */
+    remote_host = '\0';
     remote_user = 0;
 
 #if WITH_REXEC
