diff -ru e2fsprogs-1.29/config.guess e2fsprogs-1.29.new/config.guess
--- e2fsprogs-1.29/config.guess	2002-02-24 21:03:57 +0000
+++ e2fsprogs-1.29.new/config.guess	2023-07-15 15:18:53 +0000
@@ -1172,6 +1172,9 @@
     *:ITS:*:*)
 	echo pdp10-unknown-its
 	exit 0 ;;
+    i*86:Fiwix:*:*)
+	echo ${UNAME_MACHINE}-pc-fiwix
+	exit 0 ;;
 esac
 
 #echo '(No uname command or uname output not recognized.)' 1>&2
diff -ru e2fsprogs-1.29/config.sub e2fsprogs-1.29.new/config.sub
--- e2fsprogs-1.29/config.sub	2002-02-24 21:03:57 +0000
+++ e2fsprogs-1.29.new/config.sub	2023-07-15 15:19:17 +0000
@@ -1022,7 +1022,8 @@
 	      | -mingw32* | -linux-gnu* | -uxpv* | -beos* | -mpeix* | -udk* \
 	      | -interix* | -uwin* | -rhapsody* | -darwin* | -opened* \
 	      | -openstep* | -oskit* | -conix* | -pw32* | -nonstopux* \
-	      | -storm-chaos* | -tops10* | -tenex* | -tops20* | -its* | -os2*)
+	      | -storm-chaos* | -tops10* | -tenex* | -tops20* | -its* | -os2* \
+	      | -fiwix*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -ru e2fsprogs-1.29/debugfs/debugfs.c e2fsprogs-1.29.new/debugfs/debugfs.c
--- e2fsprogs-1.29/debugfs/debugfs.c	2002-07-15 20:56:47 +0000
+++ e2fsprogs-1.29.new/debugfs/debugfs.c	2023-07-15 15:19:46 +0000
@@ -21,7 +21,7 @@
 extern char *optarg;
 #endif
 #ifdef HAVE_OPTRESET
-extern int optreset;		/* defined by BSD, but not others */
+int optreset;		/* defined by BSD, but not others */
 #endif
 #ifdef HAVE_ERRNO_H
 #include <errno.h>
diff -ru e2fsprogs-1.29/lib/ext2fs/llseek.c e2fsprogs-1.29.new/lib/ext2fs/llseek.c
--- e2fsprogs-1.29/lib/ext2fs/llseek.c	2002-02-24 21:03:59 +0000
+++ e2fsprogs-1.29.new/lib/ext2fs/llseek.c	2023-07-15 15:22:12 +0000
@@ -117,14 +117,30 @@
 #define EINVAL EXT2_ET_INVALID_ARGUMENT
 #endif
 
+static ext2_loff_t my_llseek (int fd, ext2_loff_t offset, int origin)
+{
+	ext2_loff_t result;
+	int retval;
+
+	retval = llseek(fd, (unsigned long long) (offset >> 32),
+			  ((unsigned long long) offset) & 0xffffffff,
+			&result, origin);
+	return (retval == -1 ? (ext2_loff_t) retval : result);
+}
+
 ext2_loff_t ext2fs_llseek (int fd, ext2_loff_t offset, int origin)
 {
-	if ((sizeof(off_t) < sizeof(ext2_loff_t)) &&
-	    (offset >= ((ext2_loff_t) 1 << ((sizeof(off_t)*8) -1)))) {
-		errno = EINVAL;
-		return -1;
-	}
-	return lseek (fd, (off_t) offset, origin);
+	ext2_loff_t result;
+
+	if ((sizeof(off_t) >= sizeof(ext2_loff_t)) ||
+	    (offset < ((ext2_loff_t) 1 << ((sizeof(off_t)*8) -1))))
+		return lseek(fd, (off_t) offset, origin);
+
+	result = my_llseek (fd, offset, origin);
+	if (result == -1 && errno == ENOSYS) {
+ 		errno = EINVAL;
+ 	}
+	return result;
 }
 
 #endif 	/* linux */
diff -ru e2fsprogs-1.29/lib/uuid/gen_uuid.c e2fsprogs-1.29.new/lib/uuid/gen_uuid.c
--- e2fsprogs-1.29/lib/uuid/gen_uuid.c	2002-02-24 21:04:00 +0000
+++ e2fsprogs-1.29.new/lib/uuid/gen_uuid.c	2023-07-15 15:24:13 +0000
@@ -29,7 +29,9 @@
 #include <sys/stat.h>
 #include <sys/file.h>
 #include <sys/ioctl.h>
+#ifndef __fiwix__
 #include <sys/socket.h>
+#endif
 #ifdef HAVE_SYS_SOCKIO_H
 #include <sys/sockio.h>
 #endif
@@ -130,6 +132,7 @@
 #define ifreq_size(i) sizeof(struct ifreq)
 #endif /* HAVE_SA_LEN*/
 
+#ifndef __fiwix__
 	sd = socket(AF_INET, SOCK_DGRAM, IPPROTO_IP);
 	if (sd < 0) {
 		return -1;
@@ -174,6 +177,8 @@
 	close(sd);
 #endif
 	return 0;
+#endif /* ! __fiwix__ */
+	return -1;
 }
 
 /* Assume that the gettimeofday() has microsecond granularity */
diff -ru e2fsprogs-1.29/misc/mke2fs.c e2fsprogs-1.29.new/misc/mke2fs.c
--- e2fsprogs-1.29/misc/mke2fs.c	2002-09-22 19:57:59 +0000
+++ e2fsprogs-1.29.new/misc/mke2fs.c	2023-07-15 15:26:48 +0000
@@ -858,6 +858,12 @@
 		param.s_feature_ro_compat = 0;
 	}
 #endif
+#ifdef __fiwix__
+	param.s_rev_level = 0;
+	param.s_feature_incompat = 0;
+	param.s_feature_compat = 0;
+	param.s_feature_ro_compat = 0;
+#endif /* __fiwix__ */
 	fprintf (stderr, "mke2fs %s (%s)\n",
 		 E2FSPROGS_VERSION, E2FSPROGS_DATE);
 
