diff -ru gsl-2.7.1/config.guess gsl-2.7.1.new/config.guess
--- gsl-2.7.1/config.guess	2017-07-23 14:03:39 +0000
+++ gsl-2.7.1.new/config.guess	2022-04-01 21:17:44 +0000
@@ -1359,6 +1359,9 @@
     x86_64:VMkernel:*:*)
 	echo ${UNAME_MACHINE}-unknown-esx
 	exit ;;
+    i*86:Fiwix:*:*)
+	echo ${UNAME_MACHINE}-pc-fiwix
+	exit ;;
 esac
 
 eval $set_cc_for_build
diff -ru gsl-2.7.1/config.sub gsl-2.7.1.new/config.sub
--- gsl-2.7.1/config.sub	2017-07-23 14:03:39 +0000
+++ gsl-2.7.1.new/config.sub	2022-04-01 21:17:56 +0000
@@ -1374,7 +1374,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -fiwix*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
