diff -ru ncurses-examples-20180127/config.guess ncurses-examples-20180127.new/config.guess
--- ncurses-examples-20180127/config.guess	2017-11-25 21:49:31.000000000 +0100
+++ ncurses-examples-20180127.new/config.guess	2020-01-06 18:45:22.000000000 +0100
@@ -1401,6 +1401,9 @@
     amd64:Isilon\ OneFS:*:*)
 	echo x86_64-unknown-onefs
 	exit ;;
+    i*86:Fiwix:*:*)
+	echo ${UNAME_MACHINE}-pc-fiwix
+	exit ;;
 esac
 
 echo "$0: unable to guess system type" >&2
diff -ru ncurses-examples-20180127/config.sub ncurses-examples-20180127.new/config.sub
--- ncurses-examples-20180127/config.sub	2017-11-25 21:49:31.000000000 +0100
+++ ncurses-examples-20180127.new/config.sub	2020-01-06 18:46:03.000000000 +0100
@@ -1416,7 +1416,7 @@
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
-	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox*)
+	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -fiwix*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
