diff -ru ddrescue-1.28/ddrescuelog.cc ddrescue-1.28.new/ddrescuelog.cc
--- ddrescue-1.28/ddrescuelog.cc	2024-01-22 23:33:17.000000000 +0100
+++ ddrescue-1.28.new/ddrescuelog.cc	2024-08-08 15:59:45.000000000 +0200
@@ -175,7 +175,7 @@
 
 void test_offset( const long long offset, const int hardbs )
   {
-  if( std::abs( offset ) % hardbs )
+  if( llabs( offset ) % hardbs )
     { show_error( "Offset between '-i' and '-o' must be 0 or a multiple of sector size." );
       std::exit( 1 ); }
   }
