diff -ru diffutils-3.12/gnulib-tests/sys_select.in.h diffutils-3.12.new/gnulib-tests/sys_select.in.h
--- diffutils-3.12/gnulib-tests/sys_select.in.h	2025-01-12 18:13:00 +0000
+++ diffutils-3.12.new/gnulib-tests/sys_select.in.h	2025-09-13 14:30:46 +0000
@@ -97,9 +97,11 @@
    But avoid namespace pollution on glibc systems, a circular include
    <sys/select.h> -> <sys/time.h> -> <sys/select.h> on FreeBSD 13.1, and
    "unknown type name" problems on Cygwin.  */
+/*
 # if !(defined __GLIBC__ || defined __FreeBSD__ || defined __CYGWIN__)
 #  include <sys/time.h>
 # endif
+*/
 
 /* On AIX 7 and Solaris 10, <sys/select.h> provides an FD_ZERO implementation
    that relies on memset(), but without including <string.h>.
