diff -ru diffutils-3.10/gnulib-tests/sys_select.in.h diffutils-3.10.new/gnulib-tests/sys_select.in.h
--- diffutils-3.10/gnulib-tests/sys_select.in.h	2023-04-18 03:47:23.000000000 +0200
+++ diffutils-3.10.new/gnulib-tests/sys_select.in.h	2024-08-09 08:49:18.000000000 +0200
@@ -97,9 +97,11 @@
    But avoid namespace pollution on glibc systems, a circular include
    <sys/select.h> -> <sys/time.h> -> <sys/select.h> on FreeBSD 13.1, and
    "unknown type name" problems on Cygwin.  */
+/*
 # if !(defined __GLIBC__ || defined __FreeBSD__ || defined __CYGWIN__)
 #  include <sys/time.h>
 # endif
+*/
 
 /* On AIX 7 and Solaris 10, <sys/select.h> provides an FD_ZERO implementation
    that relies on memset(), but without including <string.h>.
