diff -ru dos2unix-7.5.2/Makefile dos2unix-7.5.2.new/Makefile
--- dos2unix-7.5.2/Makefile	2024-01-22 21:27:48.000000000 +0100
+++ dos2unix-7.5.2.new/Makefile	2024-08-08 09:23:44.000000000 +0200
@@ -50,9 +50,9 @@
 UNIX2MAC        = unix2mac
 
 # Native Language Support (NLS)
-ENABLE_NLS      = 1
+ENABLE_NLS      =
 # Large File Support (LFS)
-LFS             = 1
+LFS             =
 # DEBUG=1 adds the -g option to CFLAGS, for adding debug symbols.
 DEBUG = 0
 # DEBUGMSG=1 adds -DDEBUG=1 to CFLAGS, for extra verbose messages.
