diff -ru e2fsprogs-libs-1.37/config/config.guess e2fsprogs-libs-1.37.new/config/config.guess
--- e2fsprogs-libs-1.37/config/config.guess	2004-12-06 22:45:50 +0000
+++ e2fsprogs-libs-1.37.new/config/config.guess	2024-04-12 17:51:18 +0000
@@ -1260,6 +1260,9 @@
     *:DragonFly:*:*)
 	echo ${UNAME_MACHINE}-unknown-dragonfly`echo ${UNAME_RELEASE}|sed -e 's/[-(].*//'`
 	exit 0 ;;
+    i*86:Fiwix:*:*)
+	echo ${UNAME_MACHINE}-pc-fiwix
+	exit 0 ;;
 esac
 
 #echo '(No uname command or uname output not recognized.)' 1>&2
diff -ru e2fsprogs-libs-1.37/config/config.sub e2fsprogs-libs-1.37.new/config/config.sub
--- e2fsprogs-libs-1.37/config/config.sub	2004-12-06 22:45:50 +0000
+++ e2fsprogs-libs-1.37.new/config/config.sub	2024-04-12 17:51:35 +0000
@@ -1167,7 +1167,7 @@
 	      | -storm-chaos* | -tops10* | -tenex* | -tops20* | -its* \
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
-	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly*)
+	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* | -fiwix*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
