diff -ru grep-3.11/gnulib-tests/inet_pton.c grep-3.11.new/gnulib-tests/inet_pton.c
--- grep-3.11/gnulib-tests/inet_pton.c	2023-01-02 02:21:33.000000000 +0100
+++ grep-3.11.new/gnulib-tests/inet_pton.c	2024-08-09 08:38:54.000000000 +0200
@@ -32,6 +32,8 @@
  * SOFTWARE.
  */
 
+#ifndef __fiwix__
+
 #include <config.h>
 
 /* Specification.  */
@@ -266,3 +268,5 @@
 # endif
 
 #endif
+
+#endif /* ! __fiwix__ */
diff -ru grep-3.11/gnulib-tests/sys_select.in.h grep-3.11.new/gnulib-tests/sys_select.in.h
--- grep-3.11/gnulib-tests/sys_select.in.h	2023-04-18 06:44:57.000000000 +0200
+++ grep-3.11.new/gnulib-tests/sys_select.in.h	2024-08-09 08:40:41.000000000 +0200
@@ -97,9 +97,11 @@
    But avoid namespace pollution on glibc systems, a circular include
    <sys/select.h> -> <sys/time.h> -> <sys/select.h> on FreeBSD 13.1, and
    "unknown type name" problems on Cygwin.  */
+/*
 # if !(defined __GLIBC__ || defined __FreeBSD__ || defined __CYGWIN__)
 #  include <sys/time.h>
 # endif
+*/
 
 /* On AIX 7 and Solaris 10, <sys/select.h> provides an FD_ZERO implementation
    that relies on memset(), but without including <string.h>.
