diff -ru gsl-2.8/config.guess gsl-2.8.new/config.guess
--- gsl-2.8/config.guess	2017-07-23 16:03:39.000000000 +0200
+++ gsl-2.8.new/config.guess	2024-08-09 09:11:42.000000000 +0200
@@ -1356,6 +1356,9 @@
     i*86:AROS:*:*)
 	echo ${UNAME_MACHINE}-pc-aros
 	exit ;;
+    i*86:Fiwix:*:*)
+	echo ${UNAME_MACHINE}-pc-fiwix
+	exit ;;
     x86_64:VMkernel:*:*)
 	echo ${UNAME_MACHINE}-unknown-esx
 	exit ;;
diff -ru gsl-2.8/config.sub gsl-2.8.new/config.sub
--- gsl-2.8/config.sub	2017-07-23 16:03:39.000000000 +0200
+++ gsl-2.8.new/config.sub	2024-08-09 09:11:54.000000000 +0200
@@ -1374,7 +1374,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -fiwix*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
