diff -ru lcms2-2.17/config.guess lcms2-2.17.new/config.guess
--- lcms2-2.17/config.guess	2025-02-09 19:04:59 +0000
+++ lcms2-2.17.new/config.guess	2025-09-13 15:52:41 +0000
@@ -1418,6 +1418,9 @@
     amd64:Isilon\ OneFS:*:*)
 	echo x86_64-unknown-onefs
 	exit ;;
+    i*86:Fiwix:*:*)
+	echo "$UNAME_MACHINE"-pc-fiwix
+	exit ;;
 esac
 
 echo "$0: unable to guess system type" >&2
diff -ru lcms2-2.17/config.sub lcms2-2.17.new/config.sub
--- lcms2-2.17/config.sub	2025-02-09 19:04:59 +0000
+++ lcms2-2.17.new/config.sub	2025-09-13 15:52:41 +0000
@@ -1390,7 +1390,7 @@
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
 	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* \
-	      | -midnightbsd*)
+	      | -midnightbsd* | -fiwix*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -ru lcms2-2.17/utils/jpgicc/jpgicc.c lcms2-2.17.new/utils/jpgicc/jpgicc.c
--- lcms2-2.17/utils/jpgicc/jpgicc.c	2025-02-09 19:04:59 +0000
+++ lcms2-2.17.new/utils/jpgicc/jpgicc.c	2025-09-13 15:52:41 +0000
@@ -356,7 +356,6 @@
 
 typedef unsigned short uint16_t;
 typedef unsigned char uint8_t;
-typedef unsigned int uint32_t;
 
 #define INTEL_BYTE_ORDER 0x4949
 #define XRESOLUTION 0x011a
