diff -ru libarchive-3.6.2/libarchive/archive_random.c libarchive-3.6.2.new/libarchive/archive_random.c
--- libarchive-3.6.2/libarchive/archive_random.c	2022-12-09 13:38:31 +0000
+++ libarchive-3.6.2.new/libarchive/archive_random.c	2023-08-05 06:27:53 +0000
@@ -51,7 +51,7 @@
 #include <pthread.h>
 #endif
 
-static void arc4random_buf(void *, size_t);
+void arc4random_buf(void *, size_t);
 
 #endif /* HAVE_ARC4RANDOM_BUF */
 
@@ -255,7 +255,7 @@
 	return (rs.s[(si + sj) & 0xff]);
 }
 
-static void
+void
 arc4random_buf(void *_buf, size_t n)
 {
 	uint8_t *buf = (uint8_t *)_buf;
diff -ru libarchive-3.6.2/libarchive/archive_write_set_format_iso9660.c libarchive-3.6.2.new/libarchive/archive_write_set_format_iso9660.c
--- libarchive-3.6.2/libarchive/archive_write_set_format_iso9660.c	2022-12-09 13:38:31 +0000
+++ libarchive-3.6.2.new/libarchive/archive_write_set_format_iso9660.c	2023-08-05 06:29:57 +0000
@@ -63,6 +63,8 @@
 #define getgid()			0
 #endif
 
+#define timezone _timezone
+
 /*#define DEBUG 1*/
 #ifdef DEBUG
 /* To compare to the ISO image file made by mkisofs. */
