diff -ru make-4.2.1/config/config.guess make-4.2.1.new/config/config.guess
--- make-4.2.1/config/config.guess	2016-06-10 23:03:38 +0000
+++ make-4.2.1.new/config/config.guess	2022-07-15 21:54:16 +0000
@@ -1406,6 +1406,9 @@
     amd64:Isilon\ OneFS:*:*)
 	echo x86_64-unknown-onefs
 	exit ;;
+    i*86:Fiwix:*:*)
+	echo ${UNAME_MACHINE}-pc-fiwix
+	exit ;;
 esac
 
 cat >&2 <<EOF
diff -ru make-4.2.1/config/config.sub make-4.2.1.new/config/config.sub
--- make-4.2.1/config/config.sub	2016-06-10 23:03:38 +0000
+++ make-4.2.1.new/config/config.sub	2022-07-15 21:54:29 +0000
@@ -1399,7 +1399,7 @@
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
-	      | -onefs* | -tirtos* | -phoenix*)
+	      | -onefs* | -tirtos* | -phoenix* | -fiwix*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
