diff -ru musl-1.1.24/src/internal/syscall.h musl-1.1.24.new/src/internal/syscall.h
--- musl-1.1.24/src/internal/syscall.h	2019-10-13 21:58:27 +0000
+++ musl-1.1.24.new/src/internal/syscall.h	2026-07-30 16:41:50 +0000
@@ -67,6 +67,7 @@
     ((long [6]){ (long)a, (long)b, (long)c, (long)d, (long)e, (long)f }))
 #endif
 
+#ifndef __fiwix__
 /* fixup legacy 16-bit junk */
 
 #ifdef SYS_getuid32
@@ -109,7 +110,7 @@
 #define SYS_setfsuid SYS_setfsuid32
 #define SYS_setfsgid SYS_setfsgid32
 #endif
-
+#endif /* ! __fiwix__ */
 
 /* fixup legacy 32-bit-vs-lfs64 junk */
 
