diff -ru nano-8.5/src/text.c nano-8.5.new/src/text.c
--- nano-8.5/src/text.c	2025-06-08 11:26:32 +0000
+++ nano-8.5.new/src/text.c	2025-09-14 18:03:45 +0000
@@ -2133,8 +2133,7 @@
 			return;
 		}
 
-		timestamp_sec = (long)fileinfo.st_mtim.tv_sec;
-		timestamp_nsec = (long)fileinfo.st_mtim.tv_nsec;
+		timestamp_sec = (long)fileinfo.st_mtime;
 	}
 
 	/* The spell checker needs the screen, so exit from curses mode. */
@@ -2183,9 +2182,7 @@
 	free(arguments[0]);
 
 	/* When the temporary file wasn't touched, say so and leave. */
-	if (timestamp_sec > 0 && stat(tempfile_name, &fileinfo) == 0 &&
-					(long)fileinfo.st_mtim.tv_sec == timestamp_sec &&
-					(long)fileinfo.st_mtim.tv_nsec == timestamp_nsec) {
+	if ((long)fileinfo.st_mtime == timestamp_sec) {
 		statusline(REMARK, _("Nothing changed"));
 		return;
 	}
Only in nano-8.5.new/src: text.c.orig
