diff -ruN newlib-4.4.0.20231231/config.guess newlib-4.4.0.20231231.new/config.guess
--- newlib-4.4.0.20231231/config.guess	2023-12-31 17:00:18 +0000
+++ newlib-4.4.0.20231231.new/config.guess	2024-10-21 19:41:22 +0000
@@ -1494,6 +1494,9 @@
     *:Unleashed:*:*)
 	echo "$UNAME_MACHINE"-unknown-unleashed"$UNAME_RELEASE"
 	exit ;;
+    *:Fiwix:*:*)
+	echo "$UNAME_MACHINE"-pc-fiwix
+	exit ;;
 esac
 
 # No uname command or uname output not recognized.
diff -ruN newlib-4.4.0.20231231/config.sub newlib-4.4.0.20231231.new/config.sub
--- newlib-4.4.0.20231231/config.sub	2023-12-31 17:00:18 +0000
+++ newlib-4.4.0.20231231.new/config.sub	2024-10-21 19:41:22 +0000
@@ -1725,7 +1725,7 @@
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
-	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx*)
+	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | fiwix*)
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
diff -ruN newlib-4.4.0.20231231/newlib/configure newlib-4.4.0.20231231.new/newlib/configure
--- newlib-4.4.0.20231231/newlib/configure	2023-12-31 17:00:18 +0000
+++ newlib-4.4.0.20231231.new/newlib/configure	2024-10-21 19:41:22 +0000
@@ -773,6 +773,8 @@
 ENABLE_NEWLIB_ICONV_TRUE
 HAVE_LONG_DOUBLE_FALSE
 HAVE_LONG_DOUBLE_TRUE
+HAVE_LIBC_SYS_FIWIX_DIR_FALSE
+HAVE_LIBC_SYS_FIWIX_DIR_TRUE
 HAVE_LIBC_SYS_Z8KSIM_DIR_FALSE
 HAVE_LIBC_SYS_Z8KSIM_DIR_TRUE
 HAVE_LIBC_SYS_XTENSA_DIR_FALSE
@@ -5526,6 +5528,13 @@
   HAVE_LIBC_SYS_Z8KSIM_DIR_TRUE='#'
   HAVE_LIBC_SYS_Z8KSIM_DIR_FALSE=
 fi
+ if test "${sys_dir}" = fiwix; then
+  HAVE_LIBC_SYS_FIWIX_DIR_TRUE=
+  HAVE_LIBC_SYS_FIWIX_DIR_FALSE='#'
+else
+  HAVE_LIBC_SYS_FIWIX_DIR_TRUE='#'
+  HAVE_LIBC_SYS_FIWIX_DIR_FALSE=
+fi
 
 
 
@@ -7707,6 +7716,10 @@
   as_fn_error $? "conditional \"HAVE_LIBC_SYS_Z8KSIM_DIR\" was never defined.
 Usually this means the macro was only invoked conditionally." "$LINENO" 5
 fi
+if test -z "${HAVE_LIBC_SYS_FIWIX_DIR_TRUE}" && test -z "${HAVE_LIBC_SYS_FIWIX_DIR_FALSE}"; then
+  as_fn_error $? "conditional \"HAVE_LIBC_SYS_FIWIX_DIR\" was never defined.
+Usually this means the macro was only invoked conditionally." "$LINENO" 5
+fi
 if test -z "${HAVE_LONG_DOUBLE_TRUE}" && test -z "${HAVE_LONG_DOUBLE_FALSE}"; then
   as_fn_error $? "conditional \"HAVE_LONG_DOUBLE\" was never defined.
 Usually this means the macro was only invoked conditionally." "$LINENO" 5
diff -ruN newlib-4.4.0.20231231/newlib/configure.host newlib-4.4.0.20231231.new/newlib/configure.host
--- newlib-4.4.0.20231231/newlib/configure.host	2023-12-31 17:00:18 +0000
+++ newlib-4.4.0.20231231.new/newlib/configure.host	2024-10-21 19:41:22 +0000
@@ -511,6 +511,30 @@
 	unix_dir=unix
 	;;
 
+  i[34567]86-*-fiwix*)
+	sys_dir=fiwix
+	unix_dir=unix
+	posix_dir=posix
+	#crt1=crt1.o
+	#crt1_dir=libc/sys/${sys_dir}   
+	#gcc_dir=`gcc -print-search-dirs | awk '/^install:/{print $2}'`
+	default_newlib_io_c99_formats="yes"
+	default_newlib_io_long_double="yes"
+	default_newlib_io_long_long="yes"
+	default_newlib_io_pos_args="yes"
+	newlib_cflags="${newlib_cflags} -std=gnu99"
+	newlib_cflags="${newlib_cflags} -DSIGNAL_PROVIDED"
+	newlib_cflags="${newlib_cflags} -DHAVE_SYSTEM"
+	newlib_cflags="${newlib_cflags} -DHAVE_RENAME"
+	newlib_cflags="${newlib_cflags} -DHAVE_FCNTL"
+	newlib_cflags="${newlib_cflags} -DHAVE_NANOSLEEP"
+	newlib_cflags="${newlib_cflags} -DHAVE_MKDIR"
+	newlib_cflags="${newlib_cflags} -D_NO_POSIX_SPAWN"
+	newlib_cflags="${newlib_cflags} -D_I386MACH_ALLOW_HW_INTERRUPTS"        # actually disable them
+	newlib_cflags="${newlib_cflags} -D__POSIX_VISIBLE"
+	newlib_cflags="${newlib_cflags} -D__GNU_VISIBLE"
+	;;
+
   m68hc11-*-*|m6811-*-*|m6812-*-*|m68hc12-*-*)
 	;;
 
diff -ruN newlib-4.4.0.20231231/newlib/libc/acinclude.m4 newlib-4.4.0.20231231.new/newlib/libc/acinclude.m4
--- newlib-4.4.0.20231231/newlib/libc/acinclude.m4	2023-12-31 17:00:18 +0000
+++ newlib-4.4.0.20231231.new/newlib/libc/acinclude.m4	2024-10-21 19:41:22 +0000
@@ -25,6 +25,7 @@
   w65
   xtensa
   z8ksim
+  fiwix
 ], [AM_CONDITIONAL([HAVE_LIBC_SYS_]m4_toupper(SYS_DIR)[_DIR], test "${sys_dir}" = SYS_DIR)])
 
 AC_TYPE_LONG_DOUBLE
diff -ruN newlib-4.4.0.20231231/newlib/libc/include/glob.h newlib-4.4.0.20231231.new/newlib/libc/include/glob.h
--- newlib-4.4.0.20231231/newlib/libc/include/glob.h	2023-12-31 17:00:18 +0000
+++ newlib-4.4.0.20231231.new/newlib/libc/include/glob.h	2024-10-21 19:41:22 +0000
@@ -80,6 +80,7 @@
 
 #define	GLOB_NOSPACE	(-1)	/* Malloc call failed. */
 #define	GLOB_ABEND	(-2)	/* Unignored error. */
+#define	GLOB_NOMATCH	(-3)	/* No match and GLOB_NOCHECK not set. */
 
 __BEGIN_DECLS
 int	glob(const char *__restrict, int, int (*)(const char *, int), 
diff -ruN newlib-4.4.0.20231231/newlib/libc/include/sys/cdefs.h newlib-4.4.0.20231231.new/newlib/libc/include/sys/cdefs.h
--- newlib-4.4.0.20231231/newlib/libc/include/sys/cdefs.h	2023-12-31 17:00:18 +0000
+++ newlib-4.4.0.20231231.new/newlib/libc/include/sys/cdefs.h	2024-10-21 19:41:22 +0000
@@ -406,6 +406,10 @@
 #define	__unreachable()	((void)0)
 #endif
 
+#if !__GNUC_PREREQ__(5, 0)
+#define __builtin_mul_overflow(a, b, size) (*(size) = (a) * (b), 0)
+#endif
+
 /* XXX: should use `#if __STDC_VERSION__ < 199901'. */
 #if !__GNUC_PREREQ__(2, 7)
 #define	__func__	NULL
diff -ruN newlib-4.4.0.20231231/newlib/libc/posix/glob.c newlib-4.4.0.20231231.new/newlib/libc/posix/glob.c
--- newlib-4.4.0.20231231/newlib/libc/posix/glob.c	2023-12-31 17:00:18 +0000
+++ newlib-4.4.0.20231231.new/newlib/libc/posix/glob.c	2024-10-21 19:41:22 +0000
@@ -502,11 +502,14 @@
 	 * and the pattern did not contain any magic characters
 	 * GLOB_NOMAGIC is there just for compatibility with csh.
 	 */
-	if (pglob->gl_pathc == oldpathc &&
-	    ((pglob->gl_flags & GLOB_NOCHECK) ||
-	      ((pglob->gl_flags & GLOB_NOMAGIC) &&
-	       !(pglob->gl_flags & GLOB_MAGCHAR))))
-		return(globextend(pattern, pglob, limit));
+	if (pglob->gl_pathc == oldpathc) {
+		if ((pglob->gl_flags & GLOB_NOCHECK) ||
+		    ((pglob->gl_flags & GLOB_NOMAGIC) &&
+		    !(pglob->gl_flags & GLOB_MAGCHAR)))
+			return(globextend(pattern, pglob, limit));
+		else
+			return(GLOB_NOMATCH);
+	}
 	else if (!(pglob->gl_flags & GLOB_NOSORT))
 		qsort(pglob->gl_pathv + pglob->gl_offs + oldpathc,
 		    pglob->gl_pathc - oldpathc, sizeof(char *), compare);
diff -ruN newlib-4.4.0.20231231/newlib/libc/sys/Makefile.inc newlib-4.4.0.20231231.new/newlib/libc/sys/Makefile.inc
--- newlib-4.4.0.20231231/newlib/libc/sys/Makefile.inc	2023-12-31 17:00:18 +0000
+++ newlib-4.4.0.20231231.new/newlib/libc/sys/Makefile.inc	2024-10-21 19:41:22 +0000
@@ -70,3 +70,6 @@
 if HAVE_LIBC_SYS_Z8KSIM_DIR
 include %D%/z8ksim/Makefile.inc
 endif
+if HAVE_LIBC_SYS_FIWIX_DIR
+include %D%/fiwix/Makefile.inc
+endif
