diff -ru popt-1.19/build-aux/config.guess popt-1.19.new/build-aux/config.guess
--- popt-1.19/build-aux/config.guess	2022-01-04 12:08:04.000000000 +0100
+++ popt-1.19.new/build-aux/config.guess	2023-08-05 18:06:39.000000000 +0200
@@ -1418,6 +1418,9 @@
     i*86:AROS:*:*)
 	echo "$UNAME_MACHINE"-pc-aros
 	exit ;;
+    i*86:Fiwix:*:*)
+	echo "$UNAME_MACHINE"-pc-fiwix
+	exit ;;
     x86_64:VMkernel:*:*)
 	echo "$UNAME_MACHINE"-unknown-esx
 	exit ;;
diff -ru popt-1.19/build-aux/config.sub popt-1.19.new/build-aux/config.sub
--- popt-1.19/build-aux/config.sub	2022-01-04 12:08:04.000000000 +0100
+++ popt-1.19.new/build-aux/config.sub	2023-08-05 18:06:51.000000000 +0200
@@ -1360,7 +1360,7 @@
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
-	     | midnightbsd*)
+	     | midnightbsd* | fiwix*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	qnx*)
