diff -ru sed-4.7/build-aux/config.guess sed-4.7.new/build-aux/config.guess
--- sed-4.7/build-aux/config.guess	2018-12-13 07:11:38.000000000 +0100
+++ sed-4.7.new/build-aux/config.guess	2024-08-10 18:03:57.000000000 +0200
@@ -1426,6 +1426,9 @@
     *:Unleashed:*:*)
 	echo "$UNAME_MACHINE"-unknown-unleashed"$UNAME_RELEASE"
 	exit ;;
+    i*86:Fiwix:*:*)
+	echo ${UNAME_MACHINE}-pc-fiwix
+	exit ;;
 esac
 
 echo "$0: unable to guess system type" >&2
diff -ru sed-4.7/build-aux/config.sub sed-4.7.new/build-aux/config.sub
--- sed-4.7/build-aux/config.sub	2018-12-16 19:35:18.000000000 +0100
+++ sed-4.7.new/build-aux/config.sub	2024-08-10 18:04:39.000000000 +0200
@@ -1361,7 +1361,7 @@
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
-	     | midnightbsd* | amdhsa* | unleashed*)
+	     | midnightbsd* | amdhsa* | unleashed* | fiwix*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	qnx*)
diff -ru sed-4.7/gnulib-tests/sys_select.in.h sed-4.7.new/gnulib-tests/sys_select.in.h
--- sed-4.7/gnulib-tests/sys_select.in.h	2018-02-04 01:46:32.000000000 +0100
+++ sed-4.7.new/gnulib-tests/sys_select.in.h	2024-08-10 18:05:44.000000000 +0200
@@ -83,9 +83,11 @@
    in <sys/time.h>.
    But avoid namespace pollution on glibc systems and "unknown type
    name" problems on Cygwin.  */
+/*
 # if !(defined __GLIBC__ || defined __CYGWIN__)
 #  include <sys/time.h>
 # endif
+*/
 
 /* On AIX 7 and Solaris 10, <sys/select.h> provides an FD_ZERO implementation
    that relies on memset(), but without including <string.h>.
