diff -ru symlinks-1.4.3/Makefile symlinks-1.4.3.new/Makefile
--- symlinks-1.4.3/Makefile	2014-04-16 10:04:30.000000000 +0200
+++ symlinks-1.4.3.new/Makefile	2020-01-11 10:59:39.000000000 +0100
@@ -1,10 +1,11 @@
 # Makefile for symlinks
-CC     := gcc
+CC     := gcc -s
 CFLAGS += $(shell getconf LFS_CFLAGS 2>/dev/null)
 OWNER   = root
-GROUP   = root
+GROUP   = 0
+INSTALL = /usr/bin/install -D
 MANDIR  = /usr/man/man8/symlinks.8
-BINDIR  = /usr/local/bin
+BINDIR  = /usr/local/bin/symlinks
 
 .PHONY: all
 all: symlinks
