diff -ru tar-1.34/build-aux/config.guess tar-1.34.new/build-aux/config.guess
--- tar-1.34/build-aux/config.guess	2021-01-07 14:27:55 +0000
+++ tar-1.34.new/build-aux/config.guess	2022-04-10 16:15:47 +0000
@@ -1495,6 +1495,9 @@
     *:Unleashed:*:*)
 	echo "$UNAME_MACHINE"-unknown-unleashed"$UNAME_RELEASE"
 	exit ;;
+    i*86:Fiwix:*:*)
+	echo "$UNAME_MACHINE"-pc-fiwix
+	exit ;;
 esac
 
 # No uname command or uname output not recognized.
diff -ru tar-1.34/build-aux/config.sub tar-1.34.new/build-aux/config.sub
--- tar-1.34/build-aux/config.sub	2021-01-07 14:27:55 +0000
+++ tar-1.34.new/build-aux/config.sub	2022-04-10 16:15:56 +0000
@@ -1722,7 +1722,7 @@
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
-	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx*)
+	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | fiwix*)
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
diff -ru tar-1.34/lib/rtapelib.c tar-1.34.new/lib/rtapelib.c
--- tar-1.34/lib/rtapelib.c	2015-08-24 07:38:07 +0000
+++ tar-1.34.new/lib/rtapelib.c	2022-04-10 16:16:47 +0000
@@ -425,11 +425,14 @@
   }
 
   /* FIXME: Should somewhat validate the decoding, here.  */
+  /*
   if (gethostbyname (remote_host) == NULL)
     error (EXIT_ON_EXEC_ERROR, 0, _("Cannot connect to %s: resolve failed"),
 	   remote_host);
 
   if (remote_user && *remote_user == '\0')
+  */
+    remote_host = '\0';
     remote_user = 0;
 
 #if WITH_REXEC
diff -ru tar-1.34/lib/system.h tar-1.34.new/lib/system.h
--- tar-1.34/lib/system.h	2021-02-13 11:05:09 +0000
+++ tar-1.34.new/lib/system.h	2022-04-10 20:34:59 +0000
@@ -488,10 +488,12 @@
     {
       FATAL_ERROR ((0, errno, "%s(%lu)", "getpwuid", (unsigned long)uid));
     }
+/*
   if (initgroups (pw->pw_name, getgid ()))
     {
       FATAL_ERROR ((0, errno, "%s", "initgroups"));
     }
+*/
   if (gid != getegid () && setgid (gid) && errno != EPERM)
     {
       FATAL_ERROR ((0, errno, "%s", "setgid"));
diff -ru tar-1.34/lib/wordsplit.c tar-1.34.new/lib/wordsplit.c
--- tar-1.34/lib/wordsplit.c	2021-02-04 14:00:33 +0000
+++ tar-1.34.new/lib/wordsplit.c	2022-04-10 16:17:20 +0000
@@ -28,6 +28,8 @@
 #include <pwd.h>
 #include <glob.h>
 
+#define GLOB_NOMATCH	3
+
 #if ENABLE_NLS
 # include <gettext.h>
 #else
