diff -ru tree-2.1.1/Makefile tree-2.1.1.new/Makefile
--- tree-2.1.1/Makefile	2023-05-31 22:06:18.000000000 +0200
+++ tree-2.1.1.new/Makefile	2024-08-19 10:35:08.000000000 +0200
@@ -26,13 +26,13 @@
 DESTDIR=${PREFIX}/bin
 MAN=tree.1
 # Probably needs to be ${PREFIX}/share/man for most systems now
-MANDIR=${PREFIX}/man
+MANDIR=${DESTDIR}/usr/share/man
 OBJS=tree.o list.o hash.o color.o file.o filter.o info.o unix.o xml.o json.o html.o strverscmp.o
 
 # Uncomment options below for your particular OS:
 
 # Linux defaults:
-CFLAGS+=-ggdb -std=c11 -pedantic -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
+CFLAGS+=-O2 -Wall
 #CFLAGS+=-O3 -std=c11 -pedantic -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
 #LDFLAGS+=-s
 
@@ -63,7 +63,7 @@
 #CC=cc
 #CFLAGS+=-O2 -Wall -fomit-frame-pointer -no-cpp-precomp
 #LDFLAGS+=
-#MANDIR=${PREFIX}/share/man
+#MANDIR=${DESTDIR}/usr/share/man
 
 # Uncomment for HP/UX:
 #prefix=/opt
@@ -71,7 +71,7 @@
 # manpage of mbsrtowcs() requires C99 and the two defines
 #CFLAGS+=+w -D_XOPEN_SOURCE=500 -D_POSIX_C_SOURCE=200112 -AC99
 #LDFLAGS+=
-#MANDIR=${PREFIX}/share/man
+#MANDIR=${DESTDIR}/usr/share/man
 
 # Uncomment for OS/2:
 #CFLAGS+=-02 -Wall -fomit-frame-pointer -Zomf -Zsmall-conv
@@ -103,9 +103,9 @@
 	rm -f $(TREE_DEST) *.o *~
 
 install: tree
-	$(INSTALL) -d $(DESTDIR)
+	$(INSTALL) -d $(DESTDIR)/usr/bin
 	$(INSTALL) -d $(MANDIR)/man1
-	$(INSTALL) $(TREE_DEST) $(DESTDIR)/$(TREE_DEST); \
+	$(INSTALL) $(TREE_DEST) $(DESTDIR)/usr/bin/$(TREE_DEST); \
 	$(INSTALL) -m 644 doc/$(MAN) $(MANDIR)/man1/$(MAN)
 
 distclean:
