diff -ru tree-2.2.1/Makefile tree-2.2.1.new/Makefile
--- tree-2.2.1/Makefile	2024-11-26 00:55:17 +0000
+++ tree-2.2.1.new/Makefile	2025-09-15 17:16:13 +0000
@@ -27,16 +27,16 @@
 DESTDIR=${PREFIX}/bin
 MAN=tree.1
 # Probably needs to be ${PREFIX}/share/man for most systems now
-MANDIR=${PREFIX}/man
+MANDIR=${DESTDIR}/usr/share/man
 OBJS=tree.o list.o hash.o color.o file.o filter.o info.o unix.o xml.o json.o html.o strverscmp.o
 
 # Uncomment options below for your particular OS:
 
 # Linux defaults:
-LDFLAGS?=-s
+#LDFLAGS?=-s
 #CFLAGS?=-ggdb
 CFLAGS?=-O3
-CFLAGS+=-std=c11 -Wpedantic -Wall -Wextra -Wstrict-prototypes -Wshadow -Wconversion
+#CFLAGS+=-std=c11 -Wpedantic -Wall -Wextra -Wstrict-prototypes -Wshadow -Wconversion
 # _LARGEFILE64_SOURCE may be considered obsolete
 CPPFLAGS+=-DLARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
 
@@ -120,9 +120,9 @@
 	rm -f $(TREE_DEST) *.o *~
 
 install: tree
-	$(INSTALL) -d $(DESTDIR)
+	$(INSTALL) -d $(DESTDIR)/usr/bin
 	$(INSTALL) -d $(MANDIR)/man1
-	$(INSTALL) $(TREE_DEST) $(DESTDIR)/$(TREE_DEST); \
+	$(INSTALL) $(TREE_DEST) $(DESTDIR)/usr/bin/$(TREE_DEST); \
 	$(INSTALL) -m 644 doc/$(MAN) $(MANDIR)/man1/$(MAN)
 
 distclean:
