diff -ru unzip60/fileio.c unzip60.new/fileio.c
--- unzip60/fileio.c	2009-04-20 02:03:44.000000000 +0200
+++ unzip60.new/fileio.c	2024-08-19 11:10:45.000000000 +0200
@@ -1821,7 +1821,7 @@
 #else /* !(BSD || MTS || __GO32__) */
     /* tzset was already called at start of process_zipfiles() */
     /* tzset(); */              /* set `timezone' variable */
-#ifndef __BEOS__                /* BeOS DR8 has no timezones... */
+#if !defined(__BEOS__) && !defined(__fiwix__) /* BeOS DR8 has no timezones... */
     m_time += timezone;         /* seconds WEST of GMT:  add */
 #endif
 #endif /* ?(BSD || MTS || __GO32__) */
diff -ru unzip60/unix/Makefile unzip60.new/unix/Makefile
--- unzip60/unix/Makefile	2009-01-18 23:41:18.000000000 +0100
+++ unzip60.new/unix/Makefile	2024-08-19 11:09:49.000000000 +0200
@@ -123,7 +123,7 @@
 manext = 1
 prefix = /usr/local
 BINDIR = $(prefix)/bin#			where to install executables
-MANDIR = $(prefix)/man/man$(manext)#	where to install man pages
+MANDIR = $(prefix)/share/man/man$(manext)#	where to install man pages
 INSTALLEDBIN = $(BINDIR)/funzip$E $(BINDIR)/unzip$E $(BINDIR)/unzipsfx$E \
 	$(BINDIR)/zipgrep$E $(BINDIR)/zipinfo$E
 INSTALLEDMAN = $(MANDIR)/funzip.$(manext) $(MANDIR)/unzip.$(manext) \
