diff -ru util-linux-ng-2.17/config/config.guess util-linux-ng-2.17.new/config/config.guess
--- util-linux-ng-2.17/config/config.guess	2009-07-24 20:49:33 +0000
+++ util-linux-ng-2.17.new/config/config.guess	2025-07-28 14:02:22 +0000
@@ -1359,6 +1359,9 @@
     i*86:AROS:*:*)
 	echo ${UNAME_MACHINE}-pc-aros
 	exit ;;
+    i*86:Fiwix:*:*)
+	echo ${UNAME_MACHINE}-pc-fiwix
+	exit ;;
 esac
 
 #echo '(No uname command or uname output not recognized.)' 1>&2
diff -ru util-linux-ng-2.17/config/config.sub util-linux-ng-2.17.new/config/config.sub
--- util-linux-ng-2.17/config/config.sub	2009-07-24 20:49:33 +0000
+++ util-linux-ng-2.17.new/config/config.sub	2025-07-28 14:02:22 +0000
@@ -1283,7 +1283,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -fiwix*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -ru util-linux-ng-2.17/config.h.in util-linux-ng-2.17.new/config.h.in
--- util-linux-ng-2.17/config.h.in	2010-01-07 22:16:02 +0000
+++ util-linux-ng-2.17.new/config.h.in	2025-07-28 14:02:22 +0000
@@ -267,7 +267,7 @@
 /* Define to 1 if you have the `strtoull' function. */
 #undef HAVE_STRTOULL
 
-/* Define to 1 if `c_line' is member of `struct termios'. */
+/* Define to 1 if `struct termios' is a member of `c_line'. */
 #undef HAVE_STRUCT_TERMIOS_C_LINE
 
 /* Define to 1 if you have the `sysconf' function. */
@@ -400,6 +400,9 @@
 /* Define to the one symbol short name of this package. */
 #undef PACKAGE_TARNAME
 
+/* Define to the home page for this package. */
+#undef PACKAGE_URL
+
 /* Define to the version of this package. */
 #undef PACKAGE_VERSION
 
diff -ru util-linux-ng-2.17/disk-utils/Makefile.in util-linux-ng-2.17.new/disk-utils/Makefile.in
--- util-linux-ng-2.17/disk-utils/Makefile.in	2010-01-07 22:16:04 +0000
+++ util-linux-ng-2.17.new/disk-utils/Makefile.in	2025-07-28 14:39:32 +0000
@@ -39,13 +39,12 @@
 	$(srcdir)/Makefile.am $(srcdir)/Makefile.in \
 	$(top_srcdir)/config/include-Makefile.am
 @LINUX_TRUE@am__append_1 = ../lib/linux_version.c
-sbin_PROGRAMS = mkfs$(EXEEXT) mkswap$(EXEEXT) fsck.minix$(EXEEXT) \
+sbin_PROGRAMS = blockdev$(EXEEXT) mkfs$(EXEEXT) mkswap$(EXEEXT) fsck.minix$(EXEEXT) \
 	mkfs.minix$(EXEEXT) mkfs.bfs$(EXEEXT) $(am__EXEEXT_1) \
 	$(am__EXEEXT_2) $(am__EXEEXT_3) $(am__EXEEXT_4)
 usrbin_exec_PROGRAMS = isosize$(EXEEXT)
 usrsbin_exec_PROGRAMS = $(am__EXEEXT_5)
-@LINUX_TRUE@am__append_2 = fdformat.8 blockdev.8
-@LINUX_TRUE@am__append_3 = blockdev
+@LINUX_TRUE@am__append_2 = fdformat.8
 @LINUX_TRUE@am__append_4 = fdformat
 @BUILD_ELVTUNE_TRUE@am__append_5 = elvtune
 @BUILD_ELVTUNE_TRUE@am__append_6 = elvtune.8
@@ -78,7 +77,6 @@
 CONFIG_HEADER = $(top_builddir)/config.h
 CONFIG_CLEAN_FILES =
 CONFIG_CLEAN_VPATH_FILES =
-@LINUX_TRUE@am__EXEEXT_1 = blockdev$(EXEEXT)
 @BUILD_ELVTUNE_TRUE@am__EXEEXT_2 = elvtune$(EXEEXT)
 @BUILD_RAW_TRUE@@HAVE_RAW_TRUE@am__EXEEXT_3 = raw$(EXEEXT)
 @BUILD_CRAMFS_TRUE@am__EXEEXT_4 = fsck.cramfs$(EXEEXT) \
@@ -90,9 +88,9 @@
 	$(usrsbin_exec_PROGRAMS)
 am__blockdev_SOURCES_DIST = blockdev.c ../lib/blkdev.c \
 	../lib/linux_version.c
-@LINUX_TRUE@am__objects_1 = linux_version.$(OBJEXT)
+am__objects_1 = linux_version.$(OBJEXT)
 am__objects_2 = blkdev.$(OBJEXT) $(am__objects_1)
-@LINUX_TRUE@am_blockdev_OBJECTS = blockdev.$(OBJEXT) $(am__objects_2)
+am_blockdev_OBJECTS = blockdev.$(OBJEXT) $(am__objects_2)
 blockdev_OBJECTS = $(am_blockdev_OBJECTS)
 blockdev_LDADD = $(LDADD)
 AM_V_lt = $(am__v_lt_$(V))
@@ -399,7 +397,7 @@
 ul_libuuid_builddir = $(top_builddir)/shlibs/uuid/src
 ul_libuuid_la = $(top_builddir)/shlibs/uuid/src/libuuid.la
 utils_common = ../lib/blkdev.c $(am__append_1)
-dist_man_MANS = isosize.8 mkfs.8 mkswap.8 fsck.minix.8 mkfs.minix.8 \
+dist_man_MANS = blockdev.8 isosize.8 mkfs.8 mkswap.8 fsck.minix.8 mkfs.minix.8 \
 	mkfs.bfs.8 $(am__append_2) $(am__append_6) $(am__append_8)
 fsck_minix_SOURCES = fsck.minix.c minix.h
 mkfs_minix_SOURCES = mkfs.minix.c minix.h $(utils_common)
@@ -409,7 +407,7 @@
 	$(am__append_16)
 mkswap_CFLAGS = $(AM_CFLAGS) $(am__append_11) $(am__append_13) \
 	$(am__append_15)
-@LINUX_TRUE@blockdev_SOURCES = blockdev.c $(utils_common)
+blockdev_SOURCES = blockdev.c $(utils_common)
 @BUILD_ELVTUNE_TRUE@elvtune_SOURCES = elvtune.c ../lib/linux_version.c
 @BUILD_CRAMFS_TRUE@cramfs_common = $(utils_common) cramfs.h cramfs_common.c cramfs_common.h
 @BUILD_CRAMFS_TRUE@fsck_cramfs_SOURCES = fsck.cramfs.c $(cramfs_common)
diff -ru util-linux-ng-2.17/disk-utils/blockdev.c util-linux-ng-2.17.new/disk-utils/blockdev.c
--- util-linux-ng-2.17/disk-utils/blockdev.c	2009-12-01 09:31:53 +0000
+++ util-linux-ng-2.17.new/disk-utils/blockdev.c	2025-07-28 14:02:22 +0000
@@ -17,6 +17,15 @@
 
 const char *progname;
 
+#ifdef __fiwix__
+struct hd_geometry {
+	unsigned char heads;
+	unsigned char sectors;
+	unsigned short int cylinders;
+	unsigned long int start;
+};
+#endif
+
 
 struct bdc {
 	long		ioc;		/* ioctl code */
diff -ru util-linux-ng-2.17/fdisk/cfdisk.c util-linux-ng-2.17.new/fdisk/cfdisk.c
--- util-linux-ng-2.17/fdisk/cfdisk.c	2010-01-06 10:10:47 +0000
+++ util-linux-ng-2.17.new/fdisk/cfdisk.c	2025-07-28 14:02:22 +0000
@@ -160,6 +160,15 @@
 #define round_int(d) ((double)((int)(d+0.5)))
 #define ceiling(d) ((double)(((d) != (int)(d)) ? (int)(d+1.0) : (int)(d)))
 
+#ifdef __fiwix__
+struct hd_geometry {
+	unsigned char heads;
+	unsigned char sectors;
+	unsigned short int cylinders;
+	unsigned long int start;
+};
+#endif
+
 struct partition {
         unsigned char boot_ind;         /* 0x80 - active */
         unsigned char head;             /* starting head */
diff -ru util-linux-ng-2.17/fdisk/fdisk.c util-linux-ng-2.17.new/fdisk/fdisk.c
--- util-linux-ng-2.17/fdisk/fdisk.c	2010-01-06 09:59:52 +0000
+++ util-linux-ng-2.17.new/fdisk/fdisk.c	2025-07-28 14:02:22 +0000
@@ -45,6 +45,15 @@
 
 #include "gpt.h"
 
+#ifdef __fiwix__
+struct hd_geometry {
+	unsigned char heads;
+	unsigned char sectors;
+	unsigned short int cylinders;
+	unsigned long int start;
+};
+#endif
+
 static void delete_partition(int i);
 
 #define hex_val(c)	({ \
diff -ru util-linux-ng-2.17/fdisk/fdiskaixlabel.c util-linux-ng-2.17.new/fdisk/fdiskaixlabel.c
--- util-linux-ng-2.17/fdisk/fdiskaixlabel.c	2009-12-01 09:31:53 +0000
+++ util-linux-ng-2.17.new/fdisk/fdiskaixlabel.c	2025-07-28 14:02:22 +0000
@@ -7,7 +7,9 @@
 #include <string.h>             /* strstr */
 #include <unistd.h>             /* write */
 
-#include <endian.h>
+#ifdef __fiwix__
+#include <machine/endian.h>
+#endif
 
 #include "common.h"
 #include "fdisk.h"
diff -ru util-linux-ng-2.17/fdisk/fdiskmaclabel.c util-linux-ng-2.17.new/fdisk/fdiskmaclabel.c
--- util-linux-ng-2.17/fdisk/fdiskmaclabel.c	2009-12-01 09:31:53 +0000
+++ util-linux-ng-2.17.new/fdisk/fdiskmaclabel.c	2025-07-28 14:02:22 +0000
@@ -7,7 +7,9 @@
 #include <string.h>             /* strstr */
 #include <unistd.h>             /* write */
 
-#include <endian.h>
+#ifdef __fiwix__
+#include <machine/endian.h>
+#endif
 
 #include "common.h"
 #include "fdisk.h"
diff -ru util-linux-ng-2.17/fdisk/fdisksgilabel.c util-linux-ng-2.17.new/fdisk/fdisksgilabel.c
--- util-linux-ng-2.17/fdisk/fdisksgilabel.c	2009-12-01 09:31:53 +0000
+++ util-linux-ng-2.17.new/fdisk/fdisksgilabel.c	2025-07-28 14:02:22 +0000
@@ -20,7 +20,9 @@
 #include <sys/stat.h>           /* stat */
 #include <assert.h>             /* assert */
 
-#include <endian.h>
+#ifdef __fiwix__
+#include <machine/endian.h>
+#endif
 #include "nls.h"
 
 #include "blkdev.h"
@@ -29,6 +31,15 @@
 #include "fdisk.h"
 #include "fdisksgilabel.h"
 
+#ifdef __fiwix__
+struct hd_geometry {
+	unsigned char heads;
+	unsigned char sectors;
+	unsigned short int cylinders;
+	unsigned long int start;
+};
+#endif
+
 static	int     other_endian = 0;
 static	int     debug = 0;
 static  short volumes=1;
diff -ru util-linux-ng-2.17/fdisk/fdisksunlabel.c util-linux-ng-2.17.new/fdisk/fdisksunlabel.c
--- util-linux-ng-2.17/fdisk/fdisksunlabel.c	2009-12-01 09:31:53 +0000
+++ util-linux-ng-2.17.new/fdisk/fdisksunlabel.c	2025-07-28 14:02:22 +0000
@@ -21,7 +21,9 @@
 #include "nls.h"
 #include "blkdev.h"
 
-#include <endian.h>
+#ifdef __fiwix__
+#include <machine/endian.h>
+#endif
 #ifdef HAVE_SCSI_SCSI_H
 #define u_char	unsigned char
 #include <scsi/scsi.h>		/* SCSI_IOCTL_GET_IDLUN */
@@ -35,6 +37,15 @@
 #include "fdisk.h"
 #include "fdisksunlabel.h"
 
+#ifdef __fiwix__
+struct hd_geometry {
+	unsigned char heads;
+	unsigned char sectors;
+	unsigned short int cylinders;
+	unsigned long int start;
+};
+#endif
+
 static int     other_endian = 0;
 static int     scsi_disk = 0;
 static int     floppy = 0;
diff -ru util-linux-ng-2.17/fdisk/sfdisk.c util-linux-ng-2.17.new/fdisk/sfdisk.c
--- util-linux-ng-2.17/fdisk/sfdisk.c	2010-01-06 10:06:29 +0000
+++ util-linux-ng-2.17.new/fdisk/sfdisk.c	2025-07-28 14:02:22 +0000
@@ -55,6 +55,15 @@
 #include "wholedisk.h"
 #include "gpt.h"
 
+#ifdef __fiwix__
+struct hd_geometry {
+	unsigned char heads;
+	unsigned char sectors;
+	unsigned short int cylinders;
+	unsigned long int start;
+};
+#endif
+
 /*
  * Table of contents:
  *  A. About seeking
@@ -869,7 +878,7 @@
 int show_extended = 0;
 int one_only = 0;
 int one_only_pno;
-int increment = 0;
+int _increment = 0;
 
 static void
 set_format(char c) {
@@ -932,24 +941,24 @@
 	if (G.cylindersize) {
 	  printf(_("Units = cylinders of %lu bytes, blocks of 1024 bytes"
 		 ", counting from %d\n\n"),
-		 G.cylindersize<<9, increment);
+		 G.cylindersize<<9, _increment);
 	    printf(_("   Device Boot Start     End   #cyls    #blocks   Id  System\n"));
 	    break;
 	}
 	/* fall through */
       case F_SECTOR:
 	printf(_("Units = sectors of 512 bytes, counting from %d\n\n"),
-	       increment);
+	       _increment);
 	printf(_("   Device Boot    Start       End   #sectors  Id  System\n"));
 	break;
       case F_BLOCK:
 	printf(_("Units = blocks of 1024 bytes, counting from %d\n\n"),
-	       increment);
+	       _increment);
 	printf(_("   Device Boot   Start       End    #blocks   Id  System\n"));
 	break;
       case F_MEGABYTE:
 	printf(_("Units = mebibytes of 1048576 bytes, blocks of 1024 bytes"
-	       ", counting from %d\n\n"), increment);
+	       ", counting from %d\n\n"), _increment);
 	printf(_("   Device Boot Start   End    MiB    #blocks   Id  System\n"));
 	break;
     }
@@ -1065,8 +1074,8 @@
     switch(format) {
       case F_CYLINDER:
 	if (G.cylindersize) {
-	    out_rounddown(6, start, G.cylindersize, increment);
-	    out_roundup(6, end, G.cylindersize, increment);
+	    out_rounddown(6, start, G.cylindersize, _increment);
+	    out_roundup(6, end, G.cylindersize, _increment);
 	    out_roundup_size(6, size, G.cylindersize);
 	    out_rounddown(9, size, 2, 0);
 	    break;
@@ -1074,8 +1083,8 @@
 	/* fall through */
       default:
       case F_SECTOR:
-	out_rounddown(9, start, 1, increment);
-	out_roundup(9, end, 1, increment);
+	out_rounddown(9, start, 1, _increment);
+	out_roundup(9, end, 1, _increment);
 	out_rounddown(10, size, 1, 0);
 	break;
       case F_BLOCK:
@@ -1083,13 +1092,13 @@
 	printf("%8lu,%3lu ",
 	       p->sector/2, ((p->sector & 1) ? 512 : 0) + p->offset);
 #endif
-	out_rounddown(8, start, 2, increment);
-	out_roundup(8, end, 2, increment);
+	out_rounddown(8, start, 2, _increment);
+	out_roundup(8, end, 2, _increment);
 	out_rounddown(9, size, 2, 0);
 	break;
       case F_MEGABYTE:
-	out_rounddown(5, start, 2048, increment);
-	out_roundup(5, end, 2048, increment);
+	out_rounddown(5, start, 2048, _increment);
+	out_roundup(5, end, 2048, _increment);
 	out_roundup_size(5, size, 2048);
 	out_rounddown(9, size, 2, 0);
 	break;
@@ -2510,7 +2519,7 @@
 	  case 'G':
 	    opt_out_pt_geom = 1; break;
 	  case 'i':
-	    increment = 1; break;
+	    _increment = 1; break;
 	  case 'c':
 	  case 'c' + PRINT_ID:
 	  case 'c' + CHANGE_ID:
diff -ru util-linux-ng-2.17/include/bitops.h util-linux-ng-2.17.new/include/bitops.h
--- util-linux-ng-2.17/include/bitops.h	2009-11-30 15:23:17 +0000
+++ util-linux-ng-2.17.new/include/bitops.h	2025-07-28 14:02:22 +0000
@@ -2,7 +2,12 @@
 #define BITOPS_H
 
 #include <stdint.h>
+#ifndef __fiwix__
 #include <endian.h>
+#else
+#define __BYTE_ORDER	__LITTLE_ENDIAN
+#define __LITTLE_ENDIAN	1234
+#endif
 
 /*
  * Bit map related macros. Usually provided by libc.
diff -ru util-linux-ng-2.17/lib/md5.c util-linux-ng-2.17.new/lib/md5.c
--- util-linux-ng-2.17/lib/md5.c	2009-12-10 10:55:13 +0000
+++ util-linux-ng-2.17.new/lib/md5.c	2025-07-28 14:02:22 +0000
@@ -14,7 +14,12 @@
  * needed on buffers full of bytes, and then call MD5Final, which
  * will fill a supplied 16-byte array with the digest.
  */
+#ifndef __fiwix__
 #include <endian.h>
+#else
+#define __BYTE_ORDER	__LITTLE_ENDIAN
+#define __LITTLE_ENDIAN	1234
+#endif
 #include <string.h>		/* for memcpy() */
 
 #include "md5.h"
diff -ru util-linux-ng-2.17/lib/wholedisk.c util-linux-ng-2.17.new/lib/wholedisk.c
--- util-linux-ng-2.17/lib/wholedisk.c	2009-10-16 20:50:03 +0000
+++ util-linux-ng-2.17.new/lib/wholedisk.c	2025-07-28 14:02:22 +0000
@@ -6,6 +6,10 @@
 #include "blkdev.h"
 #include "wholedisk.h"
 
+#ifdef __fiwix__
+#include <fiwix/part.h>
+#endif
+
 int is_whole_disk_fd(int fd, const char *name)
 {
 #ifdef HDIO_GETGEO
diff -ru util-linux-ng-2.17/login-utils/Makefile.in util-linux-ng-2.17.new/login-utils/Makefile.in
--- util-linux-ng-2.17/login-utils/Makefile.in	2010-01-07 22:16:06 +0000
+++ util-linux-ng-2.17.new/login-utils/Makefile.in	2025-07-28 14:02:22 +0000
@@ -52,9 +52,9 @@
 @BUILD_LAST_TRUE@am__append_5 = last
 @BUILD_LAST_TRUE@am__append_6 = last.1
 @BUILD_LOGIN_UTILS_TRUE@am__append_7 = login
-@BUILD_LOGIN_UTILS_TRUE@am__append_8 = chfn chsh newgrp
-@BUILD_LOGIN_UTILS_TRUE@am__append_9 = vipw
-@BUILD_LOGIN_UTILS_TRUE@am__append_10 = chfn.1 chsh.1 login.1 newgrp.1 vipw.8 vigr.8
+#@BUILD_LOGIN_UTILS_TRUE@am__append_8 = chfn chsh newgrp
+#@BUILD_LOGIN_UTILS_TRUE@am__append_9 = vipw
+#@BUILD_LOGIN_UTILS_TRUE@am__append_10 = chfn.1 chsh.1 login.1 newgrp.1 vipw.8 vigr.8
 @BUILD_LOGIN_UTILS_TRUE@@NEED_LIBCRYPT_TRUE@am__append_11 = -lcrypt
 @BUILD_LOGIN_UTILS_TRUE@@NEED_LIBCRYPT_TRUE@am__append_12 = -lcrypt
 @BUILD_LOGIN_UTILS_TRUE@@NEED_LIBCRYPT_TRUE@am__append_13 = -lcrypt
@@ -73,7 +73,7 @@
 @BUILD_MESG_TRUE@am__append_26 = mesg.1
 @BUILD_WALL_TRUE@am__append_27 = wall
 @BUILD_WALL_TRUE@am__append_28 = wall.1
-noinst_PROGRAMS = checktty_test$(EXEEXT) islocal_test$(EXEEXT)
+#noinst_PROGRAMS = checktty_test$(EXEEXT) islocal_test$(EXEEXT)
 subdir = login-utils
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/m4/gettext.m4 \
@@ -99,11 +99,11 @@
 @BUILD_INIT_TRUE@am__EXEEXT_3 = simpleinit$(EXEEXT) shutdown$(EXEEXT) \
 @BUILD_INIT_TRUE@	initctl$(EXEEXT)
 @BUILD_LAST_TRUE@am__EXEEXT_4 = last$(EXEEXT)
-@BUILD_LOGIN_UTILS_TRUE@am__EXEEXT_5 = chfn$(EXEEXT) chsh$(EXEEXT) \
-@BUILD_LOGIN_UTILS_TRUE@	newgrp$(EXEEXT)
+#@BUILD_LOGIN_UTILS_TRUE@am__EXEEXT_5 = chfn$(EXEEXT) chsh$(EXEEXT) \
+#@BUILD_LOGIN_UTILS_TRUE@	newgrp$(EXEEXT)
 @BUILD_MESG_TRUE@am__EXEEXT_6 = mesg$(EXEEXT)
 @BUILD_WALL_TRUE@am__EXEEXT_7 = wall$(EXEEXT)
-@BUILD_LOGIN_UTILS_TRUE@am__EXEEXT_8 = vipw$(EXEEXT)
+#@BUILD_LOGIN_UTILS_TRUE@am__EXEEXT_8 = vipw$(EXEEXT)
 PROGRAMS = $(bin_PROGRAMS) $(noinst_PROGRAMS) $(sbin_PROGRAMS) \
 	$(usrbin_exec_PROGRAMS) $(usrsbin_exec_PROGRAMS)
 agetty_SOURCES = agetty.c
@@ -112,9 +112,9 @@
 AM_V_lt = $(am__v_lt_$(V))
 am__v_lt_ = $(am__v_lt_$(AM_DEFAULT_VERBOSITY))
 am__v_lt_0 = --silent
-am_checktty_test_OBJECTS = checktty_test-checktty.$(OBJEXT)
-checktty_test_OBJECTS = $(am_checktty_test_OBJECTS)
-checktty_test_LDADD = $(LDADD)
+#am_checktty_test_OBJECTS = checktty_test-checktty.$(OBJEXT)
+#checktty_test_OBJECTS = $(am_checktty_test_OBJECTS)
+#checktty_test_LDADD = $(LDADD)
 am__chfn_SOURCES_DIST = chfn.c islocal.c setpwnam.c islocal.h \
 	my_crypt.h setpwnam.h ../lib/env.c selinux_utils.c \
 	selinux_utils.h
@@ -467,8 +467,8 @@
 @BUILD_LOGIN_UTILS_TRUE@chsh_LDFLAGS = $(SUID_LDFLAGS) $(AM_LDFLAGS)
 @BUILD_LOGIN_UTILS_TRUE@newgrp_LDFLAGS = $(SUID_LDFLAGS) $(AM_LDFLAGS)
 @BUILD_WALL_TRUE@wall_SOURCES = wall.c ttymsg.c ttymsg.h
-checktty_test_SOURCES = checktty.c login.h
-checktty_test_CPPFLAGS = -DMAIN_TEST_CHECKTTY $(AM_CPPFLAGS)
+#checktty_test_SOURCES = checktty.c login.h
+#checktty_test_CPPFLAGS = -DMAIN_TEST_CHECKTTY $(AM_CPPFLAGS)
 islocal_test_SOURCES = islocal.c
 islocal_test_CPPFLAGS = -DMAIN_TEST_ISLOCAL $(AM_CPPFLAGS)
 all: all-am
@@ -689,9 +689,9 @@
 agetty$(EXEEXT): $(agetty_OBJECTS) $(agetty_DEPENDENCIES) 
 	@rm -f agetty$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(agetty_OBJECTS) $(agetty_LDADD) $(LIBS)
-checktty_test$(EXEEXT): $(checktty_test_OBJECTS) $(checktty_test_DEPENDENCIES) 
-	@rm -f checktty_test$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(checktty_test_OBJECTS) $(checktty_test_LDADD) $(LIBS)
+#checktty_test$(EXEEXT): $(checktty_test_OBJECTS) $(checktty_test_DEPENDENCIES) 
+#	@rm -f checktty_test$(EXEEXT)
+#	$(AM_V_CCLD)$(LINK) $(checktty_test_OBJECTS) $(checktty_test_LDADD) $(LIBS)
 chfn$(EXEEXT): $(chfn_OBJECTS) $(chfn_DEPENDENCIES) 
 	@rm -f chfn$(EXEEXT)
 	$(AM_V_CCLD)$(chfn_LINK) $(chfn_OBJECTS) $(chfn_LDADD) $(LIBS)
diff -ru util-linux-ng-2.17/login-utils/agetty.c util-linux-ng-2.17.new/login-utils/agetty.c
--- util-linux-ng-2.17/login-utils/agetty.c	2009-12-01 09:31:53 +0000
+++ util-linux-ng-2.17.new/login-utils/agetty.c	2025-07-28 14:02:22 +0000
@@ -29,8 +29,10 @@
 #include <getopt.h>
 #include <time.h>
 #include <sys/file.h>
+/*
 #include <sys/socket.h>
 #include <netdb.h>
+*/
 
 #include "xstrncpy.h"
 #include "nls.h"
@@ -861,6 +863,7 @@
 		   }
 		  break;
 
+/*
 		  case 'O':
 		   {
 			char *dom = "unknown_domain";
@@ -885,6 +888,7 @@
 			}
 		   }
 		  break;
+*/
 
 		  case 'd':
 		  case 't':
diff -ru util-linux-ng-2.17/login-utils/checktty.c util-linux-ng-2.17.new/login-utils/checktty.c
--- util-linux-ng-2.17/login-utils/checktty.c	2009-10-15 23:16:40 +0000
+++ util-linux-ng-2.17.new/login-utils/checktty.c	2025-07-28 14:02:22 +0000
@@ -19,8 +19,10 @@
 #include <time.h>
 #include <sys/stat.h>
 #include <malloc.h>
+/*
 #include <netdb.h>
 #include <sys/syslog.h>
+*/
 #include <ctype.h>
 #include "nls.h"
 
@@ -37,7 +39,7 @@
 #define TTY_MAJOR 4
 #endif
 
-static gid_t mygroups[NGROUPS];
+static gid_t mygroups[NGROUPS_MAX];
 static int   num_groups;
 
 #define NAMELEN 128
@@ -78,7 +80,7 @@
 static void
 find_groups(gid_t defgrp, const char *user)
 {
-	num_groups = getgroups(NGROUPS, mygroups);
+	num_groups = getgroups(NGROUPS_MAX, mygroups);
 }
 
 static struct ttyclass *
@@ -89,7 +91,7 @@
     tc = (struct ttyclass *)malloc(sizeof(struct ttyclass));
     if (tc == NULL) {
 	printf(_("login: memory low, login may fail\n"));
-	syslog(LOG_WARNING, _("can't malloc for ttyclass"));
+/*	syslog(LOG_WARNING, _("can't malloc for ttyclass"));*/
 	return NULL;
     }
 
@@ -110,7 +112,7 @@
     ge = (struct grplist *)malloc(sizeof(struct grplist));
     if (ge == NULL) {
 	printf(_("login: memory low, login may fail\n"));
-	syslog(LOG_WARNING, _("can't malloc for grplist"));
+/*	syslog(LOG_WARNING, _("can't malloc for grplist"));*/
 	return;
     }
 
@@ -159,6 +161,7 @@
 }
 
 
+#ifndef __fiwix__
 /* IPv4 -- pattern is x.x.x.x/y.y.y.y (net/mask)*/
 static int
 hnmatch_ip4(const char *pat)
@@ -281,6 +284,7 @@
 		return (strcasecmp(pat, hn + m - n) == 0);
 	}
 }
+#endif /* ! __fiwix__ */
 
 #ifdef MAIN_TEST_CHECKTTY
 
@@ -413,7 +417,7 @@
     if (strcmp(tty, class) == 0) return 1;
 
     if ((class[0] == '@') && isapty(tty)
-	&& hnmatch(hostname, class+1)) return 1;
+	/*&& hnmatch(hostname, class+1)*/) return 1;
 
     for (tc = ttyclasses; tc; tc = tc->next) {
 	if (strcmp(tc->classname, class) == 0) {
@@ -434,7 +438,7 @@
 		if (strcmp(n, tty) == 0) return 1;
 
 		if ((n[0] == '@') && isapty(tty)
-		    && hnmatch(hostname, n+1)) return 1;
+		    /*&& hnmatch(hostname, n+1)*/) return 1;
 	    }
 	    return 0;
 	}
diff -ru util-linux-ng-2.17/login-utils/login.c util-linux-ng-2.17.new/login-utils/login.c
--- util-linux-ng-2.17/login-utils/login.c	2009-10-16 20:50:03 +0000
+++ util-linux-ng-2.17.new/login-utils/login.c	2025-07-28 14:02:22 +0000
@@ -83,7 +83,7 @@
 #include <ctype.h>
 #include <unistd.h>
 #include <getopt.h>
-#include <memory.h>
+/*#include <memory.h>*/
 #include <time.h>
 #include <sys/stat.h>
 #include <sys/time.h>
@@ -102,10 +102,12 @@
 #include <utmp.h>
 #include <setjmp.h>
 #include <stdlib.h>
-#include <sys/syslog.h>
+/*#include <sys/syslog.h>*/
 #include <sys/sysmacros.h>
+/*
 #include <linux/major.h>
 #include <netdb.h>
+*/
 #ifdef HAVE_LIBAUDIT
 # include <libaudit.h>
 #endif
@@ -123,7 +125,6 @@
 #  define PAM_MAX_LOGIN_TRIES	3
 #  define PAM_FAIL_CHECK if (retcode != PAM_SUCCESS) { \
        fprintf(stderr,"\n%s\n",pam_strerror(pamh, retcode)); \
-       syslog(LOG_ERR,"%s",pam_strerror(pamh, retcode)); \
        pam_end(pamh, retcode); exit(1); \
    }
 #  define PAM_END { \
@@ -133,7 +134,7 @@
 }
 #endif
 
-#include <lastlog.h>
+/*#include <lastlog.h>*/
 
 #define SLEEP_EXIT_TIMEOUT 5
 
@@ -173,7 +174,7 @@
 static struct passwd pwdcopy;
 #endif
 char    hostaddress[16];	/* used in checktty.c */
-sa_family_t hostfamily;		/* used in checktty.c */
+/*sa_family_t hostfamily;*/		/* used in checktty.c */
 char	*hostname;		/* idem */
 static char	*username, *tty_name, *tty_number;
 static char	thishost[100];
@@ -194,8 +195,11 @@
 
 	fd = open(tty, O_RDWR | O_NONBLOCK);
 	if (fd == -1) {
+/*
 		syslog(LOG_ERR, _("FATAL: can't reopen tty: %s"),
 		       strerror(errno));
+*/
+		fprintf(stderr, "FATAL: can't reopen tty: %s.\n", strerror(errno));
 		sleep(1);
 		exit(1);
 	}
@@ -226,7 +230,8 @@
 	    || !S_ISCHR(statbuf.st_mode)
 	    || (statbuf.st_nlink > 1 && strncmp(ttyn, "/dev/", 5))
 	    || (access(ttyn, R_OK | W_OK) != 0)) {
-		syslog(LOG_ERR, _("FATAL: bad tty"));
+/*		syslog(LOG_ERR, _("FATAL: bad tty"));*/
+		fprintf(stderr, "FATAL: bad tty.\n");
 		sleep(1);
 		exit(1);
 	}
@@ -398,7 +403,7 @@
     pid = getpid();
 
     signal(SIGALRM, timedout);
-    siginterrupt(SIGALRM,1);           /* we have to interrupt syscalls like ioclt() */
+/*    siginterrupt(SIGALRM,1);*/           /* we have to interrupt syscalls like ioclt() */
     alarm((unsigned int)timeout);
     signal(SIGQUIT, SIG_IGN);
     signal(SIGINT, SIG_IGN);
@@ -407,7 +412,7 @@
     bindtextdomain(PACKAGE, LOCALEDIR);
     textdomain(PACKAGE);
 
-    setpriority(PRIO_PROCESS, 0, 0);
+/*    setpriority(PRIO_PROCESS, 0, 0);*/
     initproctitle(argc, argv);
 
     /*
@@ -443,6 +448,7 @@
 
 	  hostname = strdup(optarg); 	/* strdup: Ambrose C. Li */
 	  {
+/*
 		struct addrinfo hints, *info = NULL;
 
 		memset(&hints, 0, sizeof(hints));
@@ -466,6 +472,7 @@
 			hostfamily = info->ai_family;
 			freeaddrinfo(info);
 		}
+*/
 	  }
 	  break;
 
@@ -492,7 +499,7 @@
     } else
         ask = 1;
 
-    for (cnt = getdtablesize(); cnt > 2; cnt--)
+    for (cnt = 256 /*getdtablesize()*/; cnt > 2; cnt--)
       close(cnt);
 
     ttyn = ttyname(0);
@@ -525,7 +532,7 @@
 #endif
 
     /* set pgid to pid */
-    setpgrp();
+    setpgid(0, 0);	/*setpgrp();*/
     /* this means that setsid() will fail */
 
     {
@@ -542,7 +549,7 @@
 	/* Kill processes left on this tty */
 	tcsetattr(0,TCSAFLUSH,&ttt);
 	signal(SIGHUP, SIG_IGN); /* so vhangup() wont kill us */
-	vhangup();
+/*	vhangup();*/
 	signal(SIGHUP, SIG_DFL);
 
 	/* open stdin,stdout,stderr to the tty */
@@ -552,7 +559,7 @@
 	tcsetattr(0,TCSAFLUSH,&tt);
     }
 
-    openlog("login", LOG_ODELAY, LOG_AUTHPRIV);
+/*    openlog("login", LOG_ODELAY, LOG_AUTHPRIV);*/
 
 #if 0
     /* other than iso-8859-1 */
@@ -571,8 +578,8 @@
     if(retcode != PAM_SUCCESS) {
 	fprintf(stderr, _("%s: PAM failure, aborting: %s\n"),
 		"login", pam_strerror(pamh, retcode));
-	syslog(LOG_ERR, _("Couldn't initialize PAM: %s"),
-	       pam_strerror(pamh, retcode));
+/*	syslog(LOG_ERR, _("Couldn't initialize PAM: %s"),
+	       pam_strerror(pamh, retcode));*/
 	exit(99);
     }
     /* hostname & tty are either set to NULL or their correct values,
@@ -632,8 +639,8 @@
 	       (retcode == PAM_AUTHINFO_UNAVAIL))) {
 	    get_pam_username(pamh, &username);
 
-	    syslog(LOG_NOTICE,_("FAILED LOGIN %d FROM %s FOR %s, %s"),
-		   failcount, hostname, username, pam_strerror(pamh, retcode));
+/*	    syslog(LOG_NOTICE,_("FAILED LOGIN %d FROM %s FOR %s, %s"),
+		   failcount, hostname, username, pam_strerror(pamh, retcode));*/
 	    logbtmp(tty_name, username, hostname);
 	    logaudit(tty_name, username, hostname, NULL, 0);
 
@@ -645,6 +652,7 @@
 	if (retcode != PAM_SUCCESS) {
 	    get_pam_username(pamh, &username);
 
+/*
 	    if (retcode == PAM_MAXTRIES)
 		syslog(LOG_NOTICE,_("TOO MANY LOGIN TRIES (%d) FROM %s FOR "
 			"%s, %s"), failcount, hostname, username,
@@ -652,6 +660,7 @@
 	    else
 		syslog(LOG_NOTICE,_("FAILED LOGIN SESSION FROM %s FOR %s, %s"),
 			hostname, username, pam_strerror(pamh, retcode));
+*/
 	    logbtmp(tty_name, username, hostname);
 	    logaudit(tty_name, username, hostname, NULL, 0);
 
@@ -684,16 +693,20 @@
 
     if (!username || !*username) {
 	    fprintf(stderr, _("\nSession setup problem, abort.\n"));
+/*
 	    syslog(LOG_ERR, _("NULL user name in %s:%d. Abort."),
 		   __FUNCTION__, __LINE__);
 	    pam_end(pamh, PAM_SYSTEM_ERR);
+*/
 	    exit(1);
     }
     if (!(pwd = getpwnam(username))) {
 	    fprintf(stderr, _("\nSession setup problem, abort.\n"));
+/*
 	    syslog(LOG_ERR, _("Invalid user name \"%s\" in %s:%d. Abort."),
 		   username, __FUNCTION__, __LINE__);
 	    pam_end(pamh, PAM_SYSTEM_ERR);
+*/
 	    exit(1);
     }
 
@@ -722,12 +735,14 @@
      * This should be done before pam_setcred because
      * the PAM modules might add groups during pam_setcred.
      */
+/*
     if (initgroups(username, pwd->pw_gid) < 0) {
 	    syslog(LOG_ERR, "initgroups: %m");
 	    fprintf(stderr, _("\nSession setup problem, abort.\n"));
 	    pam_end(pamh, PAM_SYSTEM_ERR);
 	    exit(1);
     }
+*/
 
     retcode = pam_open_session(pamh, 0);
     PAM_FAIL_CHECK;
@@ -772,7 +787,7 @@
 	  salt = "xx";
 
 	if (pwd) {
-	    initgroups(username, pwd->pw_gid);
+/*	    initgroups(username, pwd->pw_gid);*/
 	    checktty(username, tty_name, pwd); /* in checktty.c */
 	}
 
@@ -801,13 +816,19 @@
 		    pwd->pw_name);
 
 	    if (hostname)
+/*
 	      syslog(LOG_NOTICE,
 		     _("LOGIN %s REFUSED FROM %s ON TTY %s"),
 		     pwd->pw_name, hostname, tty_name);
+*/
+              fprintf(stderr, "LOGIN %s REFUSED FROM %s ON TTY %s.\n", pwd->pw_name, hostname, tty_name);
 	    else
+/*
 	      syslog(LOG_NOTICE,
 		     _("LOGIN %s REFUSED ON TTY %s"),
 		     pwd->pw_name, tty_name);
+*/
+              fprintf(stderr, "LOGIN %s REFUSED ON TTY %s.\n", pwd->pw_name, tty_name);
 	    logaudit(tty_name, pwd->pw_name, hostname, pwd, 0);
 	    continue;
 	}
@@ -819,7 +840,7 @@
 	if (!passwd_req || (pwd && !*pwd->pw_passwd))
 	  break;
 
-	setpriority(PRIO_PROCESS, 0, -4);
+/*	setpriority(PRIO_PROCESS, 0, -4);*/
 	pp = getpass(_("Password: "));
 
 #  ifdef CRYPTOCARD
@@ -829,7 +850,7 @@
 #  endif /* CRYPTOCARD */
 
 	p = crypt(pp, salt);
-	setpriority(PRIO_PROCESS, 0, 0);
+/*	setpriority(PRIO_PROCESS, 0, 0);*/
 
 #  ifdef KERBEROS
 	/*
@@ -966,8 +987,10 @@
 	ut.ut_pid = pid;
 	if (hostname) {
 		xstrncpy(ut.ut_host, hostname, sizeof(ut.ut_host));
+/*
 		if (hostaddress[0])
 			memcpy(&ut.ut_addr_v6, hostaddress, sizeof(ut.ut_addr_v6));
+*/
 	}
 
 	pututline(&ut);
@@ -1084,12 +1107,16 @@
 
     setproctitle("login", username);
 
+/* Fiwix: do not report on every login.
     if (!strncmp(tty_name, "ttyS", 4))
-      syslog(LOG_INFO, _("DIALUP AT %s BY %s"), tty_name, pwd->pw_name);
+        syslog(LOG_INFO, _("DIALUP AT %s BY %s"), tty_name, pwd->pw_name);
+        fprintf(stderr, "DIALUP AT %s BY %s.\n", tty_name, pwd->pw_name);
+*/
 
     /* allow tracking of good logins.
        -steve philp (sphilp@mail.alliance.net) */
 
+/* Fiwix: do not report on every login.
     if (pwd->pw_uid == 0) {
 	if (hostname)
 	  syslog(LOG_NOTICE, _("ROOT LOGIN ON %s FROM %s"),
@@ -1104,6 +1131,7 @@
 	  syslog(LOG_INFO, _("LOGIN ON %s BY %s"), tty_name,
 		 pwd->pw_name);
     }
+*/
 
     if (!quietlog) {
 	motd();
@@ -1214,14 +1242,17 @@
     /*
      * TIOCSCTTY: steal tty from other process group.
      */
+/*
     if (ioctl(0, TIOCSCTTY, 1))
 	    syslog(LOG_ERR, _("TIOCSCTTY failed: %m"));
+*/
 #endif
     signal(SIGINT, SIG_DFL);
 
     /* discard permissions last so can't get killed and drop core */
     if(setuid(pwd->pw_uid) < 0 && pwd->pw_uid) {
-	syslog(LOG_ALERT, _("setuid() failed"));
+/*	syslog(LOG_ALERT, _("setuid() failed"));*/
+        fprintf(stderr, "setuid() failed.\n");
 	exit(1);
     }
 
@@ -1424,6 +1455,7 @@
 
 void
 dolastlog(int quiet) {
+/*
     struct lastlog ll;
     int fd;
 
@@ -1452,7 +1484,7 @@
 		time_t t;
 		time(&t);
 		ll.ll_time = t; /* ll_time is always 32bit */
-	}
+/*	}
 
 	xstrncpy(ll.ll_line, tty_name, sizeof(ll.ll_line));
 	if (hostname)
@@ -1461,10 +1493,12 @@
 	write(fd, (char *)&ll, sizeof(ll));
 	close(fd);
     }
+*/
 }
 
 void
 badlogin(const char *name) {
+/* Fiwix: do not report on every login.
     if (failures == 1) {
 	if (hostname)
 	  syslog(LOG_NOTICE, _("LOGIN FAILURE FROM %s, %s"),
@@ -1480,6 +1514,7 @@
 	  syslog(LOG_NOTICE, _("%d LOGIN FAILURES ON %s, %s"),
 		 failures, tty_name, name);
     }
+*/
 }
 
 /* Should not be called from PAM code... */
diff -ru util-linux-ng-2.17/login-utils/login.h util-linux-ng-2.17.new/login-utils/login.h
--- util-linux-ng-2.17/login-utils/login.h	2009-10-15 23:16:40 +0000
+++ util-linux-ng-2.17.new/login-utils/login.h	2025-07-28 14:02:22 +0000
@@ -3,7 +3,7 @@
 extern void sleepexit(int);
 extern char hostaddress[16];
 extern char *hostname;
-extern sa_family_t hostfamily;
+/*extern sa_family_t hostfamily;*/
 
 /* defined in checktty.c */
 extern void checktty(const char *user, const char *tty, struct passwd *pwd);
diff -ru util-linux-ng-2.17/misc-utils/findfs.c util-linux-ng-2.17.new/misc-utils/findfs.c
--- util-linux-ng-2.17/misc-utils/findfs.c	2009-10-16 20:50:03 +0000
+++ util-linux-ng-2.17.new/misc-utils/findfs.c	2025-07-28 14:02:22 +0000
@@ -16,7 +16,7 @@
 
 static void __attribute__((__noreturn__)) usage(int rc)
 {
-	const char *p = program_invocation_short_name;
+	const char *p = __progname;
 
 	if (!p)
 		p = "findfs";
diff -ru util-linux-ng-2.17/misc-utils/logger.c util-linux-ng-2.17.new/misc-utils/logger.c
--- util-linux-ng-2.17/misc-utils/logger.c	2009-10-16 20:50:03 +0000
+++ util-linux-ng-2.17.new/misc-utils/logger.c	2025-07-28 14:02:22 +0000
@@ -36,6 +36,8 @@
  * - fixed strerr(errno) in gettext calls
  */
 
+#ifndef __fiwix__
+
 #include <errno.h>
 #include <unistd.h>
 #include <stdlib.h>
@@ -285,3 +287,9 @@
 	    _("usage: logger [-is] [-f file] [-p pri] [-t tag] [-u socket] [ message ... ]\n"));
 	exit(1);
 }
+
+#endif	/* ! __fiwix__ */
+
+int main(void)
+{
+}
diff -ru util-linux-ng-2.17/misc-utils/namei.c util-linux-ng-2.17.new/misc-utils/namei.c
--- util-linux-ng-2.17/misc-utils/namei.c	2009-12-01 09:31:53 +0000
+++ util-linux-ng-2.17.new/misc-utils/namei.c	2025-07-28 14:02:22 +0000
@@ -441,7 +441,7 @@
 static void
 usage(int rc)
 {
-	const char *p = program_invocation_short_name;
+	const char *p = __progname;
 
 	if (!*p)
 		p = "namei";
diff -ru util-linux-ng-2.17/misc-utils/script.c util-linux-ng-2.17.new/misc-utils/script.c
--- util-linux-ng-2.17/misc-utils/script.c	2009-10-16 20:50:03 +0000
+++ util-linux-ng-2.17.new/misc-utils/script.c	2025-07-28 14:02:22 +0000
@@ -262,7 +262,7 @@
 	int status;
 	register int pid;
 
-	while ((pid = wait3(&status, WNOHANG, 0)) > 0)
+	while ((pid = waitpid(-1, &status, WNOHANG)) > 0)
 		if (pid == child)
 			die = 1;
 }
diff -ru util-linux-ng-2.17/misc-utils/scriptreplay.c util-linux-ng-2.17.new/misc-utils/scriptreplay.c
--- util-linux-ng-2.17/misc-utils/scriptreplay.c	2009-10-16 20:50:03 +0000
+++ util-linux-ng-2.17.new/misc-utils/scriptreplay.c	2025-07-28 14:02:22 +0000
@@ -36,7 +36,7 @@
 usage(int rc)
 {
 	printf(_("%s <timingfile> [<typescript> [<divisor>]]\n"),
-			program_invocation_short_name);
+			__progname);
 	exit(rc);
 }
 
diff -ru util-linux-ng-2.17/misc-utils/wipefs.c util-linux-ng-2.17.new/misc-utils/wipefs.c
--- util-linux-ng-2.17/misc-utils/wipefs.c	2009-12-10 11:48:53 +0000
+++ util-linux-ng-2.17.new/misc-utils/wipefs.c	2025-07-28 14:02:22 +0000
@@ -35,7 +35,7 @@
 #include "nls.h"
 
 struct wipe_desc {
-	loff_t		offset;		/* magic string offset */
+	__loff_t	offset;		/* magic string offset */
 	size_t		len;		/* length of magic string */
 	unsigned char	*magic;		/* magic string */
 
@@ -113,7 +113,7 @@
 }
 
 static struct wipe_desc *
-add_offset(struct wipe_desc *wp0, loff_t offset, int zap)
+add_offset(struct wipe_desc *wp0, __loff_t offset, int zap)
 {
 	struct wipe_desc *wp = wp0;
 
@@ -162,7 +162,7 @@
 	    blkid_probe_lookup_value(pr, "SBMAGIC", &mag, &len) == 0 &&
 	    blkid_probe_lookup_value(pr, "USAGE", &usage, NULL) == 0) {
 
-		loff_t offset = strtoll(off, NULL, 10);
+		__loff_t offset = strtoll(off, NULL, 10);
 		const char *p;
 
 		wp = add_offset(wp, offset, zap);
@@ -287,11 +287,11 @@
 	return 0;
 }
 
-static loff_t
+static __loff_t
 strtoll_offset(const char *str)
 {
 	char *end = NULL;
-	loff_t off;
+	__loff_t off;
 
 	errno = 0;
 	off = strtoll(str, &end, 0);
@@ -310,7 +310,7 @@
 usage(FILE *out)
 {
 	fprintf(out, _("Usage: %s [options] <filename>\n\nOptions:\n"),
-			program_invocation_short_name);
+			__progname);
 
 	fprintf(out, _(
 	" -a, --all           wipe all magic strings (BE CAREFUL!)\n"
diff -ru util-linux-ng-2.17/misc-utils/write.c util-linux-ng-2.17.new/misc-utils/write.c
--- util-linux-ng-2.17/misc-utils/write.c	2009-10-16 20:50:03 +0000
+++ util-linux-ng-2.17.new/misc-utils/write.c	2025-07-28 14:02:22 +0000
@@ -64,6 +64,14 @@
 #include "carefulputc.h"
 #include "nls.h"
  
+#ifndef MAXHOSTNAMELEN
+# ifdef HOST_NAME_MAX
+#  define MAXHOSTNAMELEN HOST_NAME_MAX
+# else
+#  define MAXHOSTNAMELEN 64
+# endif
+#endif
+
 void search_utmp(char *, char *, char *, uid_t);
 void do_write(char *, char *, uid_t);
 void wr_fputs(char *);
diff -ru util-linux-ng-2.17/shlibs/blkid/samples/mkfs.c util-linux-ng-2.17.new/shlibs/blkid/samples/mkfs.c
--- util-linux-ng-2.17/shlibs/blkid/samples/mkfs.c	2009-12-01 09:31:53 +0000
+++ util-linux-ng-2.17.new/shlibs/blkid/samples/mkfs.c	2025-07-28 14:02:22 +0000
@@ -27,7 +27,7 @@
 	if (argc < 2) {
 		fprintf(stderr, "usage: %s <device>  "
 			"-- checks based on libblkid for mkfs-like programs.\n",
-			program_invocation_short_name);
+			argv[0]);
 		return EXIT_FAILURE;
 	}
 
diff -ru util-linux-ng-2.17/shlibs/blkid/samples/partitions.c util-linux-ng-2.17.new/shlibs/blkid/samples/partitions.c
--- util-linux-ng-2.17/shlibs/blkid/samples/partitions.c	2009-12-01 09:31:53 +0000
+++ util-linux-ng-2.17.new/shlibs/blkid/samples/partitions.c	2025-07-28 14:02:22 +0000
@@ -26,7 +26,7 @@
 	if (argc < 2) {
 		fprintf(stderr, "usage: %s <device|file>  "
 				"-- prints partitions\n",
-				program_invocation_short_name);
+				argv[0]);
 		return EXIT_FAILURE;
 	}
 
diff -ru util-linux-ng-2.17/shlibs/blkid/samples/superblocks.c util-linux-ng-2.17.new/shlibs/blkid/samples/superblocks.c
--- util-linux-ng-2.17/shlibs/blkid/samples/superblocks.c	2009-12-01 09:31:53 +0000
+++ util-linux-ng-2.17.new/shlibs/blkid/samples/superblocks.c	2025-07-28 14:02:22 +0000
@@ -26,7 +26,7 @@
 	if (argc < 2) {
 		fprintf(stderr, "usage: %s <device>  "
 				"-- prints superblocks details about the device\n",
-				program_invocation_short_name);
+				argv[0]);
 		return EXIT_FAILURE;
 	}
 
diff -ru util-linux-ng-2.17/shlibs/blkid/samples/topology.c util-linux-ng-2.17.new/shlibs/blkid/samples/topology.c
--- util-linux-ng-2.17/shlibs/blkid/samples/topology.c	2009-12-01 09:31:53 +0000
+++ util-linux-ng-2.17.new/shlibs/blkid/samples/topology.c	2025-07-28 14:02:22 +0000
@@ -27,7 +27,7 @@
 	if (argc < 2) {
 		fprintf(stderr, "usage: %s <device>  "
 				"-- prints topology details about the device\n",
-				program_invocation_short_name);
+				argv[0]);
 		return EXIT_FAILURE;
 	}
 
diff -ru util-linux-ng-2.17/shlibs/blkid/src/blkidP.h util-linux-ng-2.17.new/shlibs/blkid/src/blkidP.h
--- util-linux-ng-2.17/shlibs/blkid/src/blkidP.h	2009-12-01 09:31:53 +0000
+++ util-linux-ng-2.17.new/shlibs/blkid/src/blkidP.h	2025-07-28 14:02:22 +0000
@@ -28,6 +28,8 @@
 #include "blkid.h"
 #include "list.h"
 
+#include <sys/sysmacros.h>	/* for major and minor */
+
 /*
  * This describes the attributes of a specific device.
  * We can traverse all of the tags by bid_tags (linking to the tag bit_names).
diff -ru util-linux-ng-2.17/sys-utils/Makefile.in util-linux-ng-2.17.new/sys-utils/Makefile.in
--- util-linux-ng-2.17/sys-utils/Makefile.in	2010-01-07 22:16:08 +0000
+++ util-linux-ng-2.17.new/sys-utils/Makefile.in	2025-07-28 14:02:22 +0000
@@ -40,7 +40,7 @@
 	$(top_srcdir)/config/include-Makefile.am
 bin_PROGRAMS = $(am__EXEEXT_1) $(am__EXEEXT_2)
 sbin_PROGRAMS = $(am__EXEEXT_3) $(am__EXEEXT_4) $(am__EXEEXT_5)
-usrbin_exec_PROGRAMS = flock$(EXEEXT) ipcrm$(EXEEXT) ipcs$(EXEEXT) \
+usrbin_exec_PROGRAMS = dmesg$(EXEEXT) flock$(EXEEXT) ipcrm$(EXEEXT) ipcs$(EXEEXT) \
 	ipcmk$(EXEEXT) renice$(EXEEXT) setsid$(EXEEXT) $(am__EXEEXT_6) \
 	$(am__EXEEXT_7) $(am__EXEEXT_8)
 usrsbin_exec_PROGRAMS = readprofile$(EXEEXT) $(am__EXEEXT_9) \
@@ -427,7 +427,7 @@
 ul_libuuid_srcdir = $(top_srcdir)/shlibs/uuid/src
 ul_libuuid_builddir = $(top_builddir)/shlibs/uuid/src
 ul_libuuid_la = $(top_builddir)/shlibs/uuid/src/libuuid.la
-dist_man_MANS = flock.1 ipcrm.1 ipcs.1 ipcmk.1 renice.1 setsid.1 \
+dist_man_MANS = dmesg.1 flock.1 ipcrm.1 ipcs.1 ipcmk.1 renice.1 setsid.1 \
 	readprofile.1 $(am__append_5) $(am__append_7) $(am__append_9) \
 	$(am__append_11) $(am__append_13) $(am__append_15) \
 	$(am__append_18)
@@ -663,6 +663,9 @@
 fallocate$(EXEEXT): $(fallocate_OBJECTS) $(fallocate_DEPENDENCIES) 
 	@rm -f fallocate$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(fallocate_OBJECTS) $(fallocate_LDADD) $(LIBS)
+dmesg$(EXEEXT): $(dmesg_OBJECTS) $(dmesg_DEPENDENCIES) 
+	@rm -f dmesg$(EXEEXT)
+	$(AM_V_CCLD)$(LINK) $(dmesg_OBJECTS) $(dmesg_LDADD) $(LIBS)
 flock$(EXEEXT): $(flock_OBJECTS) $(flock_DEPENDENCIES) 
 	@rm -f flock$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(flock_OBJECTS) $(flock_LDADD) $(LIBS)
diff -ru util-linux-ng-2.17/sys-utils/dmesg.c util-linux-ng-2.17.new/sys-utils/dmesg.c
--- util-linux-ng-2.17/sys-utils/dmesg.c	2009-10-16 20:50:03 +0000
+++ util-linux-ng-2.17.new/sys-utils/dmesg.c	2025-07-28 14:02:22 +0000
@@ -29,7 +29,7 @@
  * Only function 3 is allowed to non-root processes.
  */
 
-#include <linux/unistd.h>
+#include <fiwix/unistd.h>
 #include <stdio.h>
 #include <getopt.h>
 #include <stdlib.h>
diff -ru util-linux-ng-2.17/sys-utils/flock.c util-linux-ng-2.17.new/sys-utils/flock.c
--- util-linux-ng-2.17/sys-utils/flock.c	2009-12-07 14:23:23 +0000
+++ util-linux-ng-2.17.new/sys-utils/flock.c	2025-07-28 14:02:22 +0000
@@ -253,7 +253,7 @@
       memset(&sa, 0, sizeof sa);
 
       sa.sa_handler = timeout_handler;
-      sa.sa_flags   = SA_ONESHOT;
+      sa.sa_flags   = SA_RESETHAND;
       sigaction(SIGALRM, &sa, &old_sa);
 
       setitimer(ITIMER_REAL, &timeout, &old_timer);
diff -ru util-linux-ng-2.17/sys-utils/ipcmk.c util-linux-ng-2.17.new/sys-utils/ipcmk.c
--- util-linux-ng-2.17/sys-utils/ipcmk.c	2009-10-16 20:50:03 +0000
+++ util-linux-ng-2.17.new/sys-utils/ipcmk.c	2025-07-28 14:02:22 +0000
@@ -100,7 +100,7 @@
 	int nsems = 0;
 	int doShm = 0, doMsg = 0, doSem = 0;
 
-	progname = program_invocation_short_name;
+	progname = __progname;
 	if (!progname)
 		progname = "ipcmk";
 
diff -ru util-linux-ng-2.17/sys-utils/renice.c util-linux-ng-2.17.new/sys-utils/renice.c
--- util-linux-ng-2.17/sys-utils/renice.c	2009-10-16 20:50:03 +0000
+++ util-linux-ng-2.17.new/sys-utils/renice.c	2025-07-28 14:02:22 +0000
@@ -35,6 +35,8 @@
   * - added Native Language Support
   */
 
+#ifndef __fiwix__
+
 #include <sys/types.h>
 #include <sys/time.h>
 #include <sys/resource.h>
@@ -169,3 +171,9 @@
 	       who, oldprio, newprio);
 	return 0;
 }
+
+#endif /* !__fiwix__ */
+
+int main(void)
+{
+}
diff -ru util-linux-ng-2.17/tests/helpers/test_sysinfo.c util-linux-ng-2.17.new/tests/helpers/test_sysinfo.c
--- util-linux-ng-2.17/tests/helpers/test_sysinfo.c	2009-10-16 20:50:03 +0000
+++ util-linux-ng-2.17.new/tests/helpers/test_sysinfo.c	2025-07-28 14:02:22 +0000
@@ -18,8 +18,14 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
-#include <endian.h>
+#include <machine/endian.h>
 #include <limits.h>
+#include <bits/wordsize.h>
+
+#define __BYTE_ORDER	BYTE_ORDER
+#define __LITTLE_ENDIAN	LITTLE_ENDIAN
+#define __BIG_ENDIAN	BIG_ENDIAN
+
 
 #if !defined __BYTE_ORDER || !(__BYTE_ORDER == __LITTLE_ENDIAN) && !(__BYTE_ORDER == __BIG_ENDIAN)
 #error missing __BYTE_ORDER
diff -ru util-linux-ng-2.17/text-utils/Makefile.in util-linux-ng-2.17.new/text-utils/Makefile.in
--- util-linux-ng-2.17/text-utils/Makefile.in	2010-01-07 22:16:09 +0000
+++ util-linux-ng-2.17.new/text-utils/Makefile.in	2025-07-28 14:02:22 +0000
@@ -41,8 +41,8 @@
 usrbin_exec_PROGRAMS = col$(EXEEXT) colcrt$(EXEEXT) colrm$(EXEEXT) \
 	column$(EXEEXT) hexdump$(EXEEXT) rev$(EXEEXT) line$(EXEEXT) \
 	tailf$(EXEEXT) $(am__EXEEXT_1)
-@HAVE_NCURSES_FALSE@@HAVE_TERMCAP_TRUE@bin_PROGRAMS = more$(EXEEXT)
-@HAVE_NCURSES_TRUE@bin_PROGRAMS = more$(EXEEXT)
+#@HAVE_NCURSES_FALSE@@HAVE_TERMCAP_TRUE@bin_PROGRAMS = more$(EXEEXT)
+#@HAVE_NCURSES_TRUE@bin_PROGRAMS = more$(EXEEXT)
 @HAVE_NCURSES_TRUE@am__append_1 = ul pg
 @HAVE_NCURSES_TRUE@am__append_2 = ul.1 more.1 pg.1
 @HAVE_NCURSES_FALSE@@HAVE_TERMCAP_TRUE@am__append_3 = more.1
diff -ru util-linux-ng-2.17/text-utils/column.c util-linux-ng-2.17.new/text-utils/column.c
--- util-linux-ng-2.17/text-utils/column.c	2009-10-15 23:16:41 +0000
+++ util-linux-ng-2.17.new/text-utils/column.c	2025-07-28 14:02:22 +0000
@@ -48,6 +48,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <err.h>
+#include <termios.h>
 #include "nls.h"
 
 #include "widechar.h"
diff -ru util-linux-ng-2.17/text-utils/display.c util-linux-ng-2.17.new/text-utils/display.c
--- util-linux-ng-2.17/text-utils/display.c	2009-12-01 12:23:05 +0000
+++ util-linux-ng-2.17.new/text-utils/display.c	2025-07-28 14:02:22 +0000
@@ -48,6 +48,9 @@
 #define MIN(a,b) ((a)<(b)?(a):(b))
 #endif
 
+typedef signed long long int		quad_t;
+typedef unsigned long long int	u_quad_t;
+
 enum _vflag vflag = FIRST;
 
 static off_t address;			/* address/offset in stream */
