diff -ru ZEsarUX-13.0/generate_install_sh.sh ZEsarUX-13.0.new/generate_install_sh.sh
--- ZEsarUX-13.0/generate_install_sh.sh	2025-06-02 21:49:27 +0000
+++ ZEsarUX-13.0.new/generate_install_sh.sh	2025-10-26 08:55:03 +0000
@@ -1,7 +1,8 @@
 #!/usr/bin/env bash
 
+_INSTALL_PREFIX=`grep INSTALL_PREFIX compileoptions.h | cut -d '"' -f2`
+if [ -z "$DESTDIR" ] ; then INSTALLPREFIX="$_INSTALL_PREFIX" ; else INSTALLPREFIX="$DESTDIR/$_INSTALL_PREFIX" ; fi
 
-INSTALLPREFIX=`cat compileoptions.h |grep INSTALL_PREFIX|cut -d '"' -f2`
 EDITION_NAME_GAME=`cat cpu.h|grep EDITION_NAME_GAME|cut -d '"' -f2`
 
 
