diff -ru libuuid-1.0.3/config.guess libuuid-1.0.3.new//config.guess
--- libuuid-1.0.3/config.guess	2014-08-12 08:19:20 +0000
+++ libuuid-1.0.3.new//config.guess	2018-01-22 18:44:08 +0000
@@ -1336,6 +1336,9 @@
     x86_64:VMkernel:*:*)
 	echo ${UNAME_MACHINE}-unknown-esx
 	exit ;;
+    i*86:Fiwix:*:*)
+	echo ${UNAME_MACHINE}-pc-fiwix
+	exit ;;
 esac
 
 eval $set_cc_for_build
diff -ru libuuid-1.0.3/config.sub libuuid-1.0.3.new//config.sub
--- libuuid-1.0.3/config.sub	2014-08-12 08:19:20 +0000
+++ libuuid-1.0.3.new//config.sub	2018-01-22 18:44:20 +0000
@@ -1376,7 +1376,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -fiwix*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -ru libuuid-1.0.3/randutils.c libuuid-1.0.3.new//randutils.c
--- libuuid-1.0.3/randutils.c	2014-08-12 08:07:18 +0000
+++ libuuid-1.0.3.new//randutils.c	2018-01-22 18:51:22 +0000
@@ -13,7 +13,9 @@
 #include <string.h>
 #include <sys/time.h>
 
+#ifdef DO_JRAND_MIX
 #include <sys/syscall.h>
+#endif
 
 #include "randutils.h"
 
