diff -ru sed-4.4/build-aux/config.guess sed-4.4.new/build-aux/config.guess
--- sed-4.4/build-aux/config.guess	2017-01-17 05:35:40 +0000
+++ sed-4.4.new/build-aux/config.guess	2022-07-16 17:04:17 +0000
@@ -1412,6 +1412,9 @@
     amd64:Isilon\ OneFS:*:*)
 	echo x86_64-unknown-onefs
 	exit ;;
+    i*86:Fiwix:*:*)
+	echo ${UNAME_MACHINE}-pc-fiwix
+	exit ;;
 esac
 
 cat >&2 <<EOF
diff -ru sed-4.4/build-aux/config.sub sed-4.4.new/build-aux/config.sub
--- sed-4.4/build-aux/config.sub	2017-01-17 05:35:40 +0000
+++ sed-4.4.new/build-aux/config.sub	2022-07-16 17:04:29 +0000
@@ -1409,7 +1409,7 @@
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
-	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox*)
+	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -fiwix*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
